<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Human RNA-seq data from recount2 and related packages • recountWorkshop2020</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Human RNA-seq data from recount2 and related packages">
<meta property="og:description" content="recountWorkshop2020">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">recountWorkshop2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/recount-workshop.html">Human RNA-seq data from recount2 and related packages</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../NA/NA/NA/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Human RNA-seq data from recount2 and related packages</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                              <h4 class="author">Abhinav Nellore</h4>
            <address class="author_afil">
      Department of Biomedical Engineering, Oregon Health and Science University<br><h4 class="author">Andrew E. Jaffe</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><h4 class="date">May 29, 2019</h4>
      
      <small class="dont-index">Source: <a href="NA/NA/NA/blob/master/vignettes/recount-workshop.Rmd"><code>vignettes/recount-workshop.Rmd</code></a></small>
      <div class="hidden name"><code>recount-workshop.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="human-rna-seq-data-from-recount2-and-related-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#human-rna-seq-data-from-recount2-and-related-packages" class="anchor"></a>Human RNA-seq data from recount2 and related packages</h1>
<p>This workshop is an updated version of the <code>recountWorkshop</code> from BioC2017 and BioC2019 available at <a href="https://github.com/LieberInstitute/recountWorkshop">LieberInstitute/recountWorkshop</a> and <a href="https://github.com/LieberInstitute/recountWorkshop2019">LieberInstitute/recountWorkshop2019</a> that is adapted from the <code>recountWorkflow</code> available via <a href="https://f1000research.com/articles/6-1558/v1">F1000Research</a> and via <a href="http://www.bioconductor.org/packages/recountWorkflow">Bioconductor</a> <span class="citation">(1)</span>. This workshop uses a smaller data set to speed up computation and reduce the required computational resources to run this code.</p>
<div id="instructor" class="section level2">
<h2 class="hasAnchor">
<a href="#instructor" class="anchor"></a>Instructor</h2>
<p><a href="http://lcolladotor.github.io/">Leonardo Collado-Torres</a></p>
<ul>
<li><a href="https://github.com/lcolladotor">GitHub</a></li>
<li><a href="https://twitter.com/fellgernon">Twitter</a></li>
<li><a href="mailto:lcolladotor@gmail.com">Email</a></li>
</ul>
</div>
<div id="workshop-description" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>The recount2 project re-processed RNA sequencing (RNA-seq) data on over 70,000 human RNA-seq samples spanning a variety of tissues, cell types and disease conditions. Researchers can easily access these data via the <code>recount</code> Bioconductor package, and can quickly import gene, exon, exon-exon junction and base-pair coverage data for uniformly processed data from the SRA, GTEx and TCGA projects in R for analysis. This workshop will include an overview of the recount2 project as well as methods and tools that have been to improve it since 2017. The overview will be followed by a hands-on session where you will dive into <code>recount</code> and related packages along the lines of the <a href="http://bioconductor.org/packages/release/workflows/html/recountWorkflow.html">recount workflow</a>. We will also cover new related packages such as <a href="http://bioconductor/packages/GenomicState"><code>GenomicState</code></a>.</p>
<p>In more detail, this workshop will cover different use cases of the <code>recount</code> package, including downloading and normalizing data, processing and cleaning relevant phenotype data (including <code>recount-brain</code>), performing differential expression (DE) analyses using other Bioconductor packages. The workshop will also cover how to use the base-pair coverage data for annotation-agnostic DE analyses and for visualizing coverage data for features of interest. After taking this workshop, attendees will be ready to enhance their analyses by leveraging RNA-seq data from 70,000 human samples.</p>
<p><a href="https://jhubiostatistics.shinyapps.io/recount/">recount2 website</a>, <a href="http://bioconductor.org/packages/recount">recount package</a>, <a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html">paper</a>, <a href="http://bioconductor.org/packages/release/workflows/html/recountWorkflow.html">recount workflow</a>.</p>
</div>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Basic knowledge of RNA-seq or gene expression data</li>
<li>Familiarity with <code>SummarizedExperiment</code> <a href="http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">vignette</a>
</li>
<li>Familiarity with the <code>RangedSummarizedExperiment</code> class</li>
</ul>
</div>
<div id="workshop-participation" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h2>
<p>Students are expected to ask questions during the recount2 overview presentation and to bring their own laptop so they can follow the hands-on portion of the workshop. You can download a Docker image with all the workshop files using:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="ex">docker</span> run -e PASSWORD=bioc2020 -p 8787:8787 -d —rm lcollado/recountworkshop2020</span></code></pre></div>
<p>Then, log in to RStudio at <a href="http://localhost:8787">http://localhost:8787</a> using username <code>rstudio</code> and password <code>bioc2020</code>. Note that on Windows you need to provide your localhost IP address like <code>http://191.163.92.108:8787/</code> - find it using <code>docker-machine ip default</code> in Docker’s terminal.</p>
<p>To see the vignette on RStudio’s window (from the docker image), run <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes(package = "recountWorkshop2020")</a></code>. Click on one of the links, “HTML”, “source”, “R code”. In case of <code>The requested page was not found</code> error, add <code>help/</code> to the URL right after the hostname, e.g., <a href="http://localhost:8787/help/library/recountWorkshop2020/doc/recount-workshop.html">http://localhost:8787/help/library/recountWorkshop2020/doc/recount-workshop.html</a>. This is a <a href="https://github.com/rocker-org/rocker-versioned/issues/178">known bug</a>.</p>
<div id="bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#bioconductor-packages-used" class="anchor"></a><em>Bioconductor</em> packages used</h3>
<p><code>SummarizedExperiment</code>, <code>recount</code>, <code>GenomicRanges</code>, <code>DESeq2</code>, <code>derfinder</code>, <code>regionReport</code> are among the packages that will be explicitly covered.</p>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<colgroup>
<col width="83%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>recount2 project overview</td>
<td>10m</td>
</tr>
<tr class="even">
<td>accessing gene count data with recount</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Gene-level analysis with recount2 and recount-brain</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Un-annotated transcriptome methods overview: <code>derfinder</code>, <code>GenomicState</code>
</td>
<td>5 min</td>
</tr>
<tr class="odd">
<td>the future: recount3</td>
<td>5m</td>
</tr>
</tbody>
</table>
<p>Total: a 50-55 minute session.</p>
</div>
</div>
<div id="workshop-goals-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>Understand the publicly data available via the recount2 project beyond gene counts</li>
<li>Identify how recount2 data could be useful for your RNA-seq projects and how you can use it</li>
<li>Describe the annotation-agnostic methods powered by recount2 that complement other RNA-seq analysis pipelines</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>Re-analyze public RNA-seq data using the recount Bioconductor package</li>
<li>Recognize how to access the different types of data hosted by recount2</li>
<li>Locate resources that enhance recount2 such as recount-brain, <code>derfinder</code>, and <code>GenomicState</code>
</li>
</ul>
</div>
</div>
</div>
<div id="workshop" class="section level1">
<h1 class="hasAnchor">
<a href="#workshop" class="anchor"></a>Workshop</h1>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>RNA sequencing (RNA-seq) is now the most widely used high-throughput assay for measuring gene expression. In a typical RNA-seq experiment, several million reads are sequenced per sample. The reads are often aligned to the reference genome using a splice-aware aligner to identify where reads originated. Resulting alignment files are then used to compute count matrices which are used for several analyses such as identifying differentially expressed genes. The Bioconductor project <span class="citation">(2)</span> has many contributed packages that specialize in analyzing this type of data and previous workflows have explained how to use them <span class="citation">(3–5)</span>. Initial steps are typically focused on generating the count matrices. Some pre-computed matrices have been made available via the ReCount project <span class="citation">(6)</span> or Bioconductor Experiment data packages such as the <code>airway</code> dataset <span class="citation">(7)</span>. The count matrices in ReCount have helped others access RNA-seq data avoid having to run the processing pipelines required to compute these matrices. However, in the years since ReCount was published, hundreds of new RNA-seq projects have been carried out and researchers have publicly shared the data publicly.</p>
<p>We uniformly processed over 70,000 publicly available human RNA-seq samples and made the data available via the recount2 resource at <a href="https://jhubiostatistics.shinyapps.io/recount/">jhubiostatistics.shinyapps.io/recount/</a> <span class="citation">(8)</span>. Samples in recount2 are grouped by project (over 2,000) originating from the Sequence Read Archive, the Genotype-Tissue Expression study (GTEx) and the Cancer Genome Atlas (TCGA). The processed data can be accessed via the <code>recount</code> Bioconductor package available at <a href="http://bioconductor.org/packages/recount">bioconductor.org/packages/recount</a>. Together, recount2 and the <code>recount</code> Bioconductor package should be considered a successor to ReCount.</p>
<p>Due to space constraints, the recount2 publication <span class="citation">(8)</span> did not cover how to use the <code>recount</code> package and other useful information for carrying out analyses with recount2 data. We describe how the count matrices in recount2 were generated. We also review the R code necessary for using the recount2 data, whose details are important because some of this code involves multiple Bioconductor packages and changing default options. We further show: a) how to augment metadata that comes with datasets with metadata learned from natural language processing of associated papers as well as expression data b) how to perform differential expression analyses, and c) how to visualize the base-pair data available from recount2.</p>
<div id="recount2-overview" class="section level3">
<h3 class="hasAnchor">
<a href="#recount2-overview" class="anchor"></a>recount2 overview</h3>
<p>The recount2 resource provides expression data summarized at different feature levels to enable novel cross-study analyses. Generally when investigators use the term <em>expression</em>, they think about gene expression. But more information can be extracted from RNA-seq data. Once RNA-seq reads have been aligned to the reference genome it is possible to determine the number of aligned reads overlapping each base-pair resulting in the genome base-pair coverage curve as shown in Figure @ref(fig:Figure1). In the example shown in Figure @ref(fig:Figure1) most of the reads overlap known exons from a gene. Those reads can be used to compute a count matrix at the exon or gene feature levels. Some reads span exon-exon junctions (jx) and while most match the annotation, some do not (jx 3 and 4). An exon-exon junction count matrix can be used to identify differentially expressed junctions, which can show which isoforms are differentially expressed given sufficient coverage. For example, junctions 2 and 5 are unique to isoform 2, while junction 6 is unique to isoform 1. The genome base-pair coverage data can be used with <code>derfinder</code> <span class="citation">(9)</span> to identify expressed regions, some of them could be un-annotated exons and together with the exon-exon junction data uncover potential new isoforms.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure1.png" alt="Overview of the data available in recount2. Reads (pink boxes) aligned to the reference genome can be used to compute a base-pair coverage curve and identify exon-exon junctions (split reads). Gene and exon count matrices are generated using annotation information providing the gene (green boxes) and exon (blue boxes) coordinates together with the base-level coverage curve. The reads spanning exon-exon junctions (jx) are used to compute a third count matrix that might include un-annotated junctions (jx 3 and 4). Without using annotation information, expressed regions (orange box) can be determined from the base-level coverage curve to then construct data-driven count matrices." width="100%"><p class="caption">
Overview of the data available in recount2. Reads (pink boxes) aligned to the reference genome can be used to compute a base-pair coverage curve and identify exon-exon junctions (split reads). Gene and exon count matrices are generated using annotation information providing the gene (green boxes) and exon (blue boxes) coordinates together with the base-level coverage curve. The reads spanning exon-exon junctions (jx) are used to compute a third count matrix that might include un-annotated junctions (jx 3 and 4). Without using annotation information, expressed regions (orange box) can be determined from the base-level coverage curve to then construct data-driven count matrices.
</p>
</div>
<p>recount2 provides gene, exon, and exon-exon junction count matrices both in text format and <em>RangedSummarizedExperiment</em> objects (RSE) <span class="citation">(10)</span> as shown in Figure @ref(fig:Figure2). These RSE objects provide information about the expression features (for example gene ids) and the samples. In this workshop we will explain how to add metadata to the RSE objects in recount2 in order to ask biological questions. recount2 also provides coverage data in the form of BigWig files. All four features can be accessed with the <code>recount</code> Bioconductor package <span class="citation">(8)</span>. <code>recount</code> also allows sending queries to <code>snaptron</code> <span class="citation">(11)</span> to search for specific exon-exon junctions.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure2.png" alt="recount2 provides coverage count matrices in RangedSummarizedExperiment (rse) objects. Once the rse object has been downloaded and loaded into R, the feature information is accessed with rowRanges(rse) (blue box), the counts with assays(rse)\$counts (pink box) and the sample metadata with colData(rse) (green box). The sample metadata can be expanded using add\_predictions(rse) (orange box) or with custom code (brown box) matching by a unique sample identifier such as the SRA Run id. The rse object is inside the purple box and matching data is highlighted in each box." width="100%"><p class="caption">
recount2 provides coverage count matrices in RangedSummarizedExperiment (rse) objects. Once the rse object has been downloaded and loaded into R, the feature information is accessed with rowRanges(rse) (blue box), the counts with assays(rse)$counts (pink box) and the sample metadata with colData(rse) (green box). The sample metadata can be expanded using add_predictions(rse) (orange box) or with custom code (brown box) matching by a unique sample identifier such as the SRA Run id. The rse object is inside the purple box and matching data is highlighted in each box.
</p>
</div>
</div>
<div id="packages-used-in-the-workshop" class="section level3">
<h3 class="hasAnchor">
<a href="#packages-used-in-the-workshop" class="anchor"></a>Packages used in the workshop</h3>
<p>In this workshop we will use several Bioconductor packages. To reproduce the entirety of this workshop, install the packages using the following code after installing R 4.0 from CRAN in order to use Bioconductor version 3.11 or newer.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">## Install packages from Bioconductor</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"recount"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"DESeq2"</span>, <span class="st">"ideal"</span>, <span class="st">"regionReport"</span>,
    <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"gplots"</span>, <span class="st">"derfinder"</span>,
    <span class="st">"rtracklayer"</span>, <span class="st">"GenomicFeatures"</span>, <span class="st">"bumphunter"</span>, <span class="st">"derfinderPlot"</span>,
    <span class="st">"sessioninfo"</span>, <span class="st">"GenomicState"</span>
))
</pre></div>
<p>Once installed, load all of the packages with the following code.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/recount">"recount"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/mikelove/DESeq2">"DESeq2"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/federicomarini/ideal">"ideal"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/regionReport">"regionReport"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://guangchuangyu.github.io/software/clusterProfiler">"clusterProfiler"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"org.Hs.eg.db"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/talgalili/gplots">"gplots"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/lcolladotor/derfinder">"derfinder"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"rtracklayer"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicFeatures"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/ririzarr/bumphunter">"bumphunter"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/derfinderPlot">"derfinderPlot"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/LieberInstitute/GenomicState">"GenomicState"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/r-lib/sessioninfo#readme">"sessioninfo"</a></span>)
</pre></div>
</div>
</div>
<div id="coverage-counts-provided-by-recount2" class="section level2">
<h2 class="hasAnchor">
<a href="#coverage-counts-provided-by-recount2" class="anchor"></a>Coverage counts provided by recount2</h2>
<p>The most accessible features are the gene, exon and exon-exon junction count matrices. This section explains them in greater detail. Figure @ref(fig:Figure3) shows 16 RNA-seq reads each 3 base-pairs long and a reference genome.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure3.png" alt="RNA-seq starting data. 16 RNA-seq un-aligned RNA-seq reads 3 base-pairs long are shown (pink boxes) along a reference genome 16 base-pairs long (white box)." width="100%"><p class="caption">
RNA-seq starting data. 16 RNA-seq un-aligned RNA-seq reads 3 base-pairs long are shown (pink boxes) along a reference genome 16 base-pairs long (white box).
</p>
</div>
<p>Reads in the recount2 resource were aligned with Rail-RNA aligner <span class="citation">(12)</span> which is splice-aware and can soft clip reads. Figure @ref(fig:Figure4) shows the reads aligned to the reference genome. Some of the reads are split as they span an exon-exon junction. Two of the reads were soft clipped meaning that just a portion of the reads aligned (top left in purple).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure4.png" alt="Aligned RNA-seq reads. Spice-aware RNA-seq aligners such as Rail-RNA are able to find the coordinates to which the reads map, even if they span exon-exon junctions (connected boxes). Rail-RNA soft clips some reads (purple boxes with rough edges) such that a portion of these reads align to the reference genome." width="100%"><p class="caption">
Aligned RNA-seq reads. Spice-aware RNA-seq aligners such as Rail-RNA are able to find the coordinates to which the reads map, even if they span exon-exon junctions (connected boxes). Rail-RNA soft clips some reads (purple boxes with rough edges) such that a portion of these reads align to the reference genome.
</p>
</div>
<p>In order to compute the gene and exon count matrices we first have to process the annotation, which for recount2 is Gencode v25 (CHR regions) with hg38 coordinates. Although <code>recount</code> can generate count matrices for other annotations using hg38 coordinates. Figure @ref(fig:Figure5) shows two isoforms for a gene composed of 3 different exons.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure5.png" alt="Gene annotation. A single gene with two isoforms composed by three distinct exons (blue boxes) is illustrated. Exons 1 and 3 share the first five base-pairs while exon 2 is common to both isoforms." width="100%"><p class="caption">
Gene annotation. A single gene with two isoforms composed by three distinct exons (blue boxes) is illustrated. Exons 1 and 3 share the first five base-pairs while exon 2 is common to both isoforms.
</p>
</div>
<p>The coverage curve is at the base-pair resolution, so if we are interested in gene counts we have to be careful not to double count base-pairs 1 through 5 that are shared by exons 1 and 3 (Figure @ref(fig:Figure5)). Using the function <code><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">disjoin()</a></code> from <code>GenomicRanges</code> <span class="citation">(13)</span> we identified the distinct exonic sequences (disjoint exons). The following code defines the exon coordinates that match Figure @ref(fig:Figure5) and the resulting disjoint exons for our example gene. The resulting disjoint exons are shown in Figure @ref(fig:Figure6).</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="kw">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(<span class="st">"seq"</span>, <span class="fu">IRanges</span>(start = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">13</span>), end = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5</span>, <span class="fl">8</span>, <span class="fl">15</span>)))
<span class="kw">exons</span>
</pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]      seq       1-5      *
##   [2]      seq       1-8      *
##   [3]      seq     13-15      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">disjoin</a></span>(<span class="kw">exons</span>)
</pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]      seq       1-5      *
##   [2]      seq       6-8      *
##   [3]      seq     13-15      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure6.png" alt="Disjoint exons. Windows of distinct exonic sequence for the example gene. Disjoint exons 1 and 2 form exon 1." width="100%"><p class="caption">
Disjoint exons. Windows of distinct exonic sequence for the example gene. Disjoint exons 1 and 2 form exon 1.
</p>
</div>
<p>Now that we have disjoint exons, we can compute the base-pair coverage for each of them as shown in Figure @ref(fig:Figure7). That is, for each base-pair that corresponds to exonic sequence, we compute the number of reads overlapping that given base-pair. For example, the first base-pair is covered by 3 different reads and it does not matter whether the reads themselves were soft clipped. Not all reads or bases of a read contribute information to this step as some do not overlap known exonic sequence (light pink in Figure @ref(fig:Figure7)).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure7.png" alt="Base-pair coverage counting for exonic base-pairs. At each exonic base-pair we compute the number of reads overlapping that given base-pair. The first base (orange arrow) has 3 reads overlapping that base-pair. Base-pair 11 has a coverage of 3 but does not overlap known exonic sequence, so that information is not used for the gene and exon count matrices (grey arrow). If a read partially overlaps exonic sequence, only the portion that overlaps is used in the computation (see right most read)." width="100%"><p class="caption">
Base-pair coverage counting for exonic base-pairs. At each exonic base-pair we compute the number of reads overlapping that given base-pair. The first base (orange arrow) has 3 reads overlapping that base-pair. Base-pair 11 has a coverage of 3 but does not overlap known exonic sequence, so that information is not used for the gene and exon count matrices (grey arrow). If a read partially overlaps exonic sequence, only the portion that overlaps is used in the computation (see right most read).
</p>
</div>
<p>With base-pair coverage for the exonic sequences computed, the coverage count for each distinct exon is simply the sum of the base-pair coverage for each base in a given distinct exon. For example, the coverage count for disjoint exon 2 is <span class="math inline">\(2 + 2 + 3 = 7\)</span> as shown in Figure @ref(fig:Figure8). The gene coverage count is then <span class="math inline">\(\sum_i^n \texttt{coverage}_i\)</span> where <span class="math inline">\(n\)</span> is the number of exonic base-pairs for the gene and is equal to the sum of the coverage counts for its disjoint exons as shown in Figure @ref(fig:Figure8).</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure8.png" alt="Exon and gene coverage counts. The coverage counts for each disjoint exon are the sum of the base-pair coverage. The gene coverage count is the sum of the disjoint exons coverage counts." width="100%"><p class="caption">
Exon and gene coverage counts. The coverage counts for each disjoint exon are the sum of the base-pair coverage. The gene coverage count is the sum of the disjoint exons coverage counts.
</p>
</div>
<p>For the exons, recount2 provides the disjoint exons coverage count matrix. It is possible to reconstruct the exon coverage count matrix by summing the coverage count for the disjoint exons that compose each exon. For example, the coverage count for exon 1 would be the sum of the coverage counts for disjoint exons 1 and 2, that is <span class="math inline">\(19 + 7 = 26\)</span>. Some methods might assume that double counting of the shared base-pairs was performed while others assume or recommend the opposite.</p>
<div id="scaling-coverage-counts" class="section level3">
<h3 class="hasAnchor">
<a href="#scaling-coverage-counts" class="anchor"></a>Scaling coverage counts</h3>
<p>The coverage counts described previously are the ones actually included in the rse objects in recount2 instead of typical read count matrices. This is an important difference to keep in mind as most methods were developed for read count matrices. Part of the sample metadata available from recount2 includes the read length and number of mapped reads. Given a target library size (40 million reads by default), the coverage counts in recount2 can be scaled to read counts for a given library size as shown in Equation @ref(eq:scale). Note that the resulting scaled read counts are not necessarily integers so it might be neccessary to round them if a differential expression method assumes integer data.</p>
<p><span class="math display">\[\begin{equation}
  \frac{\sum_i^n \text{coverage}_i }{\text{Read Length}} * \frac{\text{target}}{\text{mapped}} = \text{scaled read counts}
  (\#eq:scale)
\end{equation}\]</span></p>
<p>From Figure @ref(fig:Figure4) we know that Rail-RNA soft clipped some reads, so a more precise measure than the denominator of Equation @ref(eq:scale) is the area under coverage (AUC) which is the sum of the coverage for all base-pairs of the genome, regardless of the annotation as shown in Figure @ref(fig:Figure9). Without soft clipping reads, the AUC would be equal to the number of reads mapped multiplied by the read length. So for our example gene, the scaled counts for a library size of 20 reads would be <span class="math inline">\(\frac{36}{45} * 20 = 16\)</span> and in general calculated with Equation @ref(eq:scale2). The following code shows how to compute the AUC given a set of aligned reads and reproduce a portion of Figure @ref(fig:Figure9).</p>
<p><span class="math display">\[\begin{equation}
  \frac{\sum_i^n \text{coverage}_i }{\text{AUC}} * \text{target} = \text{scaled read counts}
  (\#eq:scale2)
\end{equation}\]</span></p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co">## Take the example and translate it to R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)
<span class="kw">reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(<span class="st">"seq"</span>, <span class="fu">IRanges</span>(
    start = <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">7</span>, <span class="fl">8</span>, <span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">13</span>, <span class="fl">14</span>),
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">1</span>)
    ), width = <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">3</span>),
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>)
    )
))
<span class="co">## Get the base-level genome coverage curve</span>
<span class="kw">cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/coverage-methods.html">coverage</a></span>(<span class="kw">reads</span>)<span class="op">$</span><span class="kw">seq</span>)

<span class="co">## AUC</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">cov</span>)
</pre></div>
<pre><code>## [1] 45</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co">## Code for reproducing the bottom portion of Figure 8.</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"base_pair_coverage.pdf"</span>, width = <span class="fl">20</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mar = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5</span>, <span class="fl">6</span>, <span class="fl">4</span>, <span class="fl">2</span>) <span class="op">+</span> <span class="fl">0.1</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="kw">cov</span>,
    type = <span class="st">"o"</span>, col = <span class="st">"violetred1"</span>, lwd = <span class="fl">10</span>, ylim = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">5</span>),
    xlab = <span class="st">"Genome"</span>, ylab = <span class="st">"Coverage"</span>, cex.axis = <span class="fl">2</span>, cex.lab = <span class="fl">3</span>,
    bty = <span class="st">"n"</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/polygon.html">polygon</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">cov</span>)), <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">cov</span>)), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="kw">cov</span>, <span class="fl">0</span>),
    border = <span class="fl">NA</span>, density = <span class="op">-</span><span class="fl">1</span>, col = <span class="st">"light blue"</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">cov</span>)), <span class="kw">cov</span>,
    col = <span class="st">"violetred1"</span>, type = <span class="st">"o"</span>,
    lwd = <span class="fl">10</span>
)
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()
</pre></div>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure9.png" alt="Area under coverage (AUC). The area under coverage is the sum of the base-pair coverage for all positions in the genome regardless of the annotation. It is the area under the base-level coverage curve shown as the light blue area under the pink curve." width="100%"><p class="caption">
Area under coverage (AUC). The area under coverage is the sum of the base-pair coverage for all positions in the genome regardless of the annotation. It is the area under the base-level coverage curve shown as the light blue area under the pink curve.
</p>
</div>
<p>The <code>recount</code> function <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code> computes the scaled read counts for a target library size of 40 million mapped reads and we highly recommend using it before doing other analyses. The following code shows how to use <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code> and that the resulting read counts per sample can be lower than the target size (40 million). This happens when not all mapped reads overlap known exonic base-pairs of the genome. In our example, the gene has a scaled count of 16 reads for a library size of 20 reads, meaning that 4 reads did not overlap exonic sequences.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co">## Check that the number of reads is less than or equal to 40 million</span>
<span class="co">## after scaling.</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/recount">"recount"</a></span>)
<span class="kw">rse_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="kw">rse_gene_SRP009615</span>, round = <span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(<span class="fu">assays</span>(<span class="kw">rse_scaled</span>)<span class="op">$</span><span class="kw">counts</span>)) <span class="op">/</span> <span class="fl">1e6</span>
</pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   22.62   29.97   34.00   31.96   34.86   36.78</code></pre>
</div>
<div id="enriching-the-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#enriching-the-annotation" class="anchor"></a>Enriching the annotation</h3>
<p>Data in recount2 can be used for annotation-agnostic analyses and enriching the known annotation. Just like exon and gene coverage count matrices, recount2 provides exon-exon junction count matrices. These matrices can be used to identify new isoforms (Figure @ref(fig:Figure10)) or identify differentially expressed isoforms. For example, exon-exon junctions 2, 5 and 6 in Figure @ref(fig:Figure1) are only present in one annotated isoform. <code>Snaptron</code> <span class="citation">(11)</span> allows programatic and high-level queries of the exon-exon junction information and it is graphical user interface is specially useful for visualizing this data. Inside R, the <code>recount</code> function <code><a href="https://rdrr.io/pkg/recount/man/snaptron_query.html">snaptron_query()</a></code> can be used for searching specific exon-exon junctions in recount2. A full R interface to <code>Snaptron</code> is under development.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure10.png" alt="Exon-exon junctions go beyond the annotation. Reads spanning exon-exon junctions are highlighted and compared against the annotation. Three of them match the annotated junctions, but one (blue and orange read) spans an un-annotated exon-exon junction with the left end matching the annotation and the right end hinting at a possible new isoform for this gene (blue and orange isoform)." width="100%"><p class="caption">
Exon-exon junctions go beyond the annotation. Reads spanning exon-exon junctions are highlighted and compared against the annotation. Three of them match the annotated junctions, but one (blue and orange read) spans an un-annotated exon-exon junction with the left end matching the annotation and the right end hinting at a possible new isoform for this gene (blue and orange isoform).
</p>
</div>
<p>The base-pair coverage data from recount2 can be used together with <code>derfinder</code> <span class="citation">(9)</span> to identify expressed regions of the genome, providing another annotation-agnostic analysis of the expression data. Using the function <code><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html">expressed_regions()</a></code> we can identify regions of expression based on a given data set in recount2. These regions might overlap known exons but can also provide information about intron retention events (Figure @ref(fig:Figure11)), improve detection of exon boundaries (Figure @ref(fig:Figure12)), and help identify new exons (Fig @ref(fig:Figure1)) or expressed sequences in intergenic regions. Using <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code> we can compute a coverage matrix based on the expressed regions or another set of genomic intervals. The resulting matrix can then be used for a differential expression analysis, just like the exon, gene and exon-exon junction matrices.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure11.png" alt="Intron retention events. Some reads might align with known intronic segments of the genome and provide information for exploring intron retention events (pink read). Some might support an intron retention event or a new isoform when coupled with exon-exon junction data (orange read)." width="100%"><p class="caption">
Intron retention events. Some reads might align with known intronic segments of the genome and provide information for exploring intron retention events (pink read). Some might support an intron retention event or a new isoform when coupled with exon-exon junction data (orange read).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure12.png" alt="Exon boundaries. Reads that go beyond the known exon boundaries can inform us of whether the annotated boundaries are correct or if there was a run-off transcription event." width="100%"><p class="caption">
Exon boundaries. Reads that go beyond the known exon boundaries can inform us of whether the annotated boundaries are correct or if there was a run-off transcription event.
</p>
</div>
</div>
</div>
<div id="gene-level-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-level-analysis" class="anchor"></a>Gene level analysis</h2>
<p>Having reviewed how the coverage counts in recount2 were produced, we can now do a differential expression analysis. We will use data from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67333">GSE67333</a> <span class="citation">(14)</span> whose overall design was: <em>We performed directional RNA sequencing on high quality RNA samples extracted from hippocampi of 4 late onset Alzheimer’s disease (LOAD) and 4 age-matched controls.</em> The function <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code> requires a SRA accession id which can be found using <code><a href="https://rdrr.io/pkg/recount/man/abstract_search.html">abstract_search()</a></code>. <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code> can then be used to download the gene coverage count data as well as other expression features. The files are saved in a directory named after the SRA accession id, in this case SRP056604.</p>
<p>To facilitate this workshop, we included the data in the <code>recountWorkshop2020</code> package. It can be located with the <code><a href="https://rdrr.io/r/base/system.file.html">system.file()</a></code> function as shown below.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co">## Locate path with the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com//LieberInstitute/recountWorkshop2020">"recountWorkshop2020"</a></span>)
<span class="kw">local_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"SRP056604"</span>, package = <span class="st">"recountWorkshop2020"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="kw">local_path</span>)
</pre></div>
<pre><code>## [1] "bw"              "rse_exon.Rdata"  "rse_gene.Rdata"  "SraRunTable.txt"</code></pre>
<p>The commands should work with the data hosted from recount2 if there are no problems with the wifi connection. Although we prefer that you use the pre-installed data for this workshop.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/recount">"recount"</a></span>)

<span class="co">## Find the project id by searching abstracts of studies</span>
<span class="fu"><a href="https://rdrr.io/pkg/recount/man/abstract_search.html">abstract_search</a></span>(<span class="st">"hippocampi of 4"</span>)
</pre></div>
<pre><code>##      number_samples species
## 1643              8   human
##                                                                                                                                                                                                                                                                                                                                                                                                    abstract
## 1643 Our data provide a comprehensive list of transcriptomics alterations and warrant holistic approach including both coding and non-coding RNAs in functional studies aimed to understand the pathophysiology of LOAD Overall design: We performed directional RNA sequencing on high quality RNA samples extracted from hippocampi of 4 late onset Alzheimer's diseas (LOAD) and 4 age-matched controls.
##        project
## 1643 SRP056604</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co">## Download the data if it is not there</span>
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"rse_gene.Rdata"</span>))) {
    <span class="co">## In case you decide to download the data instead of using the</span>
    <span class="co">## pre-installed data</span>
    <span class="kw">local_path</span> <span class="op">&lt;-</span> <span class="st">"SRP056604"</span>
    <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span>(<span class="st">"SRP056604"</span>, type = <span class="st">"rse-gene"</span>)
}

<span class="co">## Check that the file was downloaded</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"rse_gene.Rdata"</span>))
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co">## Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"rse_gene.Rdata"</span>), verbose = <span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Loading objects:
##   rse_gene</code></pre>
<p>The coverage count matrices are provided as <em>RangedSummarizedExperiment</em> objects (RSE) <span class="citation">(10)</span>. These objects store information at the feature level, the samples and the actual count matrix as shown in Figure 1 of Love et al., 2016 <span class="citation">(4)</span>. Figure @ref(fig:Figure2) shows the actual RSE objects provided by recount2 and how to access the different portions of the data. Using a unique sample id such as the SRA Run id it is possible to expand the sample metadata. This can be done using the predicted phenotype provided by <code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions()</a></code> <span class="citation">(15)</span>, pulling information from GEO via <code><a href="https://rdrr.io/pkg/recount/man/find_geo.html">find_geo()</a></code> and <code><a href="https://rdrr.io/pkg/recount/man/geo_characteristics.html">geo_characteristics()</a></code>, or with custom code. As of 2019, you can also use <code><a href="https://rdrr.io/pkg/recount/man/add_metadata.html">add_metadata(source = "recount_brain_v2")</a></code> to add curated sample metadata from brain studies <span class="citation">(16)</span>.</p>
<div id="metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#metadata" class="anchor"></a>Metadata</h3>
<p>Using the <code>colData()</code> function we can access sample metadata. More information on these metadata is provided in the supplementary material of the recount2 paper <span class="citation">(8)</span>, and we provide a brief review here. The RSE objects for SRA data sets include 21 columns with mostly technical information. The GTEx and TCGA RSE objects include additional metadata as available from the raw sources. In particular, we compiled metadata for GTEx using the v6 phenotype information available at <a href="gtexportal.org">gtexportal.org</a>, and we put together a large table of TCGA case and sample information by combining information accumulated across <a href="http://www.cancergenomicscloud.org/">Seven Bridges’ Cancer Genomics Cloud</a> and <code>TCGAbiolinks</code> <span class="citation">(17)</span>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="co">## One row per sample, one column per phenotype variable</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>## [1]  8 21</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="co">## Mostly technical variables are included</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>##  [1] "project"                                       
##  [2] "sample"                                        
##  [3] "experiment"                                    
##  [4] "run"                                           
##  [5] "read_count_as_reported_by_sra"                 
##  [6] "reads_downloaded"                              
##  [7] "proportion_of_reads_reported_by_sra_downloaded"
##  [8] "paired_end"                                    
##  [9] "sra_misreported_paired_end"                    
## [10] "mapped_read_count"                             
## [11] "auc"                                           
## [12] "sharq_beta_tissue"                             
## [13] "sharq_beta_cell_type"                          
## [14] "biosample_submission_date"                     
## [15] "biosample_publication_date"                    
## [16] "biosample_update_date"                         
## [17] "avg_read_length"                               
## [18] "geo_accession"                                 
## [19] "bigwig_file"                                   
## [20] "title"                                         
## [21] "characteristics"</code></pre>
<div id="technical-variables" class="section level4">
<h4 class="hasAnchor">
<a href="#technical-variables" class="anchor"></a>Technical variables</h4>
<p>Several of these technical variables include the number of reads as reported by SRA, the actual number of reads Rail-RNA was able to download (which might be lower in some cases), the number of reads mapped by Rail-RNA, whether the sample is paired-end or not, the coverage AUC and the average read length (times 2 for paired-end samples). Note that sample with SRA Run id SRR2071341 has about 240.8 million reads as reported by SRA, while it has 120.4 million spots reported in <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR2071341" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR2071341</a>; that is because it is a paired-end sample (2 reads per spot). These details are important for those interested on writing alternative scaling functions to <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts()</a></code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="co">## Input reads: number reported by SRA might be larger than number</span>
<span class="co">## of reads Rail-RNA downloaded</span>
<span class="kw">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/all_metadata.html">all_metadata</a></span>()
</pre></div>
<pre><code>## 2020-07-26 21:24:06 downloading the metadata to /tmp/RtmpKDI1FD/metadata_clean_sra.Rdata</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="kw">meta</span>[
    <span class="kw">meta</span><span class="op">$</span><span class="kw">run</span> <span class="op">==</span> <span class="st">"SRR2071341"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"read_count_as_reported_by_sra"</span>, <span class="st">"reads_downloaded"</span>)
]
</pre></div>
<pre><code>## DataFrame with 1 row and 2 columns
##   read_count_as_reported_by_sra reads_downloaded
##                       &lt;integer&gt;        &lt;integer&gt;
## 1                     240797206        240797206</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="kw">meta</span><span class="op">$</span><span class="kw">proportion_of_reads_reported_by_sra_downloaded</span>[
    <span class="kw">meta</span><span class="op">$</span><span class="kw">project</span> <span class="op">==</span> <span class="st">"SRP045638"</span>
])
</pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.5719  0.9165  0.9788  0.9532  1.0000  1.0000</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="co">## AUC information used by scale_counts() by default</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">auc</span>)
</pre></div>
<pre><code>## [1] 14058249889 13996129504 13578647527 14599138400 16423440640 15559854293</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="co">## Alternatively, scale_counts() can use the number of mapped reads</span>
<span class="co">## and other information</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mapped_read_count"</span>, <span class="st">"paired_end"</span>, <span class="st">"avg_read_length"</span>)]
</pre></div>
<pre><code>## DataFrame with 8 rows and 3 columns
##            mapped_read_count paired_end avg_read_length
##                    &lt;integer&gt;  &lt;logical&gt;       &lt;integer&gt;
## SRR1931819         160584474      FALSE              99
## SRR1931818         155722526      FALSE              99
## SRR1931817         164110375      FALSE              99
## SRR1931816         165119594      FALSE              99
## SRR1931815         180961318      FALSE              99
## SRR1931814         170579381      FALSE              99
## SRR1931813         171525236      FALSE              99
## SRR1931812         178912737      FALSE              99</code></pre>
</div>
<div id="biological-information" class="section level4">
<h4 class="hasAnchor">
<a href="#biological-information" class="anchor"></a>Biological information</h4>
<p>Other metadata variables included provide more biological information, such as the <a href="http://www.cs.cmu.edu/~ckingsf/sharq/">SHARQ beta</a> tissue and cell type predictions, which are based on processing the abstract of papers. This information is available for some of the SRA projects.</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="co">## SHARQ tissue predictions: not present for all studies</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sharq_beta_tissue</span>)
</pre></div>
<pre><code>## [1] NA NA NA NA NA NA</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene_SRP009615</span>)<span class="op">$</span><span class="kw">sharq_beta_tissue</span>)
</pre></div>
<pre><code>## [1] "blood" "blood" "blood" "blood" "blood" "blood"</code></pre>
<p>For some data sets we were able to find the GEO accession ids, which we then used to create the <code>title</code> and <code>characteristics</code> variables. If present, the <code>characteristics</code> information can be used to create additional metadata variables by parsing the <code>CharacterList</code> in which it is stored. Since the input is free text, sometimes more than one type of wording is used to describe the same information. Meaning that we might have to process that information in order to build a more convenient variable, such as a factor vector.</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="co">## GEO information was absent for the SRP056604 data set</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"geo_accession"</span>, <span class="st">"title"</span>, <span class="st">"characteristics"</span>)]
</pre></div>
<pre><code>## DataFrame with 8 rows and 3 columns
##            geo_accession       title
##              &lt;character&gt; &lt;character&gt;
## SRR1931819    GSM1645003       CTRL4
## SRR1931818    GSM1645002       CTRL3
## SRR1931817    GSM1645001       CTRL2
## SRR1931816    GSM1645000       CTRL1
## SRR1931815    GSM1644999       LOAD4
## SRR1931814    GSM1644998       LOAD3
## SRR1931813    GSM1644997       LOAD2
## SRR1931812    GSM1644996       LOAD1
##                                                    characteristics
##                                                    &lt;CharacterList&gt;
## SRR1931819 age (yrs): 85,subject group: age-m..,gender: female,...
## SRR1931818  age (yrs): &gt;90,subject group: age-m..,gender: male,...
## SRR1931817   age (yrs): 83,subject group: age-m..,gender: male,...
## SRR1931816 age (yrs): 77,subject group: age-m..,gender: female,...
## SRR1931815 age (yrs): 82,subject group: LOAD ..,gender: female,...
## SRR1931814   age (yrs): 83,subject group: LOAD ..,gender: male,...
## SRR1931813 age (yrs): 87,subject group: LOAD ..,gender: female,...
## SRR1931812 age (yrs): 82,subject group: LOAD ..,gender: female,...</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="co">## GEO information for the SRP009615 data set</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene_SRP009615</span>)<span class="op">$</span><span class="kw">geo_accession</span>)
</pre></div>
<pre><code>## [1] "GSM836270" "GSM836271" "GSM836272" "GSM836273" "GSM847561" "GSM847562"</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene_SRP009615</span>)<span class="op">$</span><span class="kw">title</span>, <span class="fl">2</span>)
</pre></div>
<pre><code>## [1] "K562 cells with shRNA targeting SRF gene cultured with no doxycycline (uninduced - UI), rep1." 
## [2] "K562 cells with shRNA targeting SRF gene cultured with doxycycline for 48 hours (48 hr), rep1."</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene_SRP009615</span>)<span class="op">$</span><span class="kw">characteristics</span>, <span class="fl">2</span>)
</pre></div>
<pre><code>## CharacterList of length 2
## [[1]] cells: K562 shRNA expression: no treatment: Puromycin
## [[2]] cells: K562 shRNA expression: yes, targeting SRF treatment: Puromycin, doxycycline</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="co">## Similar but not exactly the same wording used for two different samples</span>
<span class="fu">colData</span>(<span class="kw">rse_gene_SRP009615</span>)<span class="op">$</span><span class="kw">characteristics</span>[[<span class="fl">1</span>]]
</pre></div>
<pre><code>## [1] "cells: K562"          "shRNA expression: no" "treatment: Puromycin"</code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="fu">colData</span>(<span class="kw">rse_gene_SRP009615</span>)<span class="op">$</span><span class="kw">characteristics</span>[[<span class="fl">11</span>]]
</pre></div>
<pre><code>## [1] "cell line: K562"                      
## [2] "shRNA expression: no shRNA expression"
## [3] "treatment: Puromycin"</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="co">## For study SRP056604 we have characteristics information</span>
<span class="co">## Note the &gt; symbol in the age for the second sample</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">characteristics</span>[[<span class="fl">1</span>]]
</pre></div>
<pre><code>## [1] "age (yrs): 85"                      "subject group: age-matched control"
## [3] "gender: female"                     "apoe genotype: 2/3"                
## [5] "braak stage: II"                    "tissue: Hippocampus"</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">characteristics</span>[[<span class="fl">2</span>]]
</pre></div>
<pre><code>## [1] "age (yrs): &gt;90"                     "subject group: age-matched control"
## [3] "gender: male"                       "apoe genotype: 3/3"                
## [5] "braak stage: II"                    "tissue: Hippocampus"</code></pre>
<p>Since we have the <code>characteristics</code> information for study SRP056604, lets extract some data from them. We can extract the <code>case</code> status, <code>sex</code> and <code>age</code>.</p>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="co">## Get the case status: either LOAD or control</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">case</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">characteristics</span>, <span class="fu">function</span>(<span class="kw">x</span>) {
          <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"LOAD"</span>, <span class="kw">x</span>)), <span class="st">"LOAD"</span>, <span class="st">"control"</span>)
      })
)

<span class="co">## Get the sex</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">characteristics</span>, <span class="fu">function</span>(<span class="kw">x</span>) {
          <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"female"</span>, <span class="kw">x</span>)),
              <span class="st">"female"</span>, <span class="st">"male"</span>
          )
      })
)

<span class="co">## Extract the age. Note that one of them is a bit more complicated.</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(
    <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">characteristics</span>,
    <span class="fu">function</span>(<span class="kw">x</span>) {
        <span class="kw">y</span> <span class="op">&lt;-</span> <span class="kw">x</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"age.*(yrs)"</span>, <span class="kw">x</span>)]
        <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"age.*\\(yrs\\): |&gt;"</span>, <span class="st">""</span>, <span class="kw">y</span>))
    }
)
</pre></div>
<p>Now that we have added the biological metadata variables, we can explore them.</p>
<div class="sourceCode" id="cb56"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">case</span>, <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sex</span>)
</pre></div>
<pre><code>##          
##           female male
##   control      2    2
##   LOAD         3    1</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">case</span>, <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">age</span>)
</pre></div>
<pre><code>##          
##           77 82 83 85 87 90
##   control  1  0  1  1  0  1
##   LOAD     0  2  1  0  1  0</code></pre>
<p>From the previous tables, case seems to be balanced by sex and age.</p>
<p>As shown in Figure @ref(fig:Figure2), we can expand the biological metadata information by adding predictions based on RNA-seq data <span class="citation">(15)</span>. The predictions include information about sex, sample source (cell line vs tissue), tissue and the sequencing strategy used. To add the predictions, simply use the function <code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions()</a></code> to expand the <code>colData()</code> slot.</p>
<div class="sourceCode" id="cb60"><pre class="downlit">
<span class="co">## Before adding predictions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>## [1]  8 24</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit">
<span class="co">## Add the predictions</span>
<span class="kw">rse_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions</a></span>(<span class="kw">rse_gene</span>)
</pre></div>
<pre><code>## 2020-07-26 21:24:10 downloading the predictions to /tmp/RtmpKDI1FD/PredictedPhenotypes_v0.0.06.rda</code></pre>
<pre><code>## Loading objects:
##   PredictedPhenotypes</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit">
<span class="co">## After adding the predictions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>## [1]  8 36</code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit">
<span class="co">## Explore the variables</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)[, <span class="fl">25</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))]
</pre></div>
<pre><code>## DataFrame with 8 rows and 12 columns
##            reported_sex predicted_sex accuracy_sex reported_samplesource
##                &lt;factor&gt;      &lt;factor&gt;    &lt;numeric&gt;              &lt;factor&gt;
## SRR1931819           NA        male       0.862637                tissue
## SRR1931818           NA        male       0.862637                tissue
## SRR1931817           NA        female     0.862637                tissue
## SRR1931816           NA        female     0.862637                tissue
## SRR1931815           NA        female     0.862637                tissue
## SRR1931814           NA        female     0.862637                tissue
## SRR1931813           NA        male       0.862637                tissue
## SRR1931812           NA        female     0.862637                tissue
##            predicted_samplesource accuracy_samplesource reported_tissue
##                          &lt;factor&gt;             &lt;numeric&gt;        &lt;factor&gt;
## SRR1931819                 tissue               0.89235              NA
## SRR1931818                 tissue               0.89235              NA
## SRR1931817                 tissue               0.89235              NA
## SRR1931816                 tissue               0.89235              NA
## SRR1931815                 tissue               0.89235              NA
## SRR1931814                 tissue               0.89235              NA
## SRR1931813                 tissue               0.89235              NA
## SRR1931812                 tissue               0.89235              NA
##            predicted_tissue accuracy_tissue reported_sequencingstrategy
##                    &lt;factor&gt;       &lt;numeric&gt;                    &lt;factor&gt;
## SRR1931819            Brain        0.518825                      SINGLE
## SRR1931818            Brain        0.518825                      SINGLE
## SRR1931817            Brain        0.518825                      SINGLE
## SRR1931816            Brain        0.518825                      SINGLE
## SRR1931815            Brain        0.518825                      SINGLE
## SRR1931814            Brain        0.518825                      SINGLE
## SRR1931813            Brain        0.518825                      SINGLE
## SRR1931812            Brain        0.518825                      SINGLE
##            predicted_sequencingstrategy accuracy_sequencingstrategy
##                                &lt;factor&gt;                   &lt;numeric&gt;
## SRR1931819                       SINGLE                    0.908575
## SRR1931818                       SINGLE                    0.908575
## SRR1931817                       SINGLE                    0.908575
## SRR1931816                       SINGLE                    0.908575
## SRR1931815                       SINGLE                    0.908575
## SRR1931814                       SINGLE                    0.908575
## SRR1931813                       SINGLE                    0.908575
## SRR1931812                       SINGLE                    0.908575</code></pre>
<p>From the predictions, we see that all samples are predicted to be from the brain, which matches the samples description. We also have predicted sex which we can compare against the reported sex.</p>
<div class="sourceCode" id="cb69"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(
    <span class="st">"Observed"</span> = <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sex</span>,
    <span class="st">"Predicted"</span> = <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">predicted_sex</span>
)
</pre></div>
<pre><code>##         Predicted
## Observed female male Unassigned
##   female      3    2          0
##   male        2    1          0</code></pre>
<p>We have 4 sex prediction mismatches, which is very high. Let’s explore whether the sex is matching is related to other variables.</p>
<div class="sourceCode" id="cb71"><pre class="downlit">
<span class="co">## Is the sex matching? TRUE for yes.</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">matching_sex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sex</span>) <span class="op">==</span>
    <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">predicted_sex</span>)

<span class="co">## Matching sex vs other variables</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="st">"Matched Sex"</span> = <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">matching_sex</span>, <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">case</span>)
</pre></div>
<pre><code>##            
## Matched Sex control LOAD
##       FALSE       2    2
##       TRUE        2    2</code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="st">"Matched Sex"</span> = <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">matching_sex</span>, <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">age</span>)
</pre></div>
<pre><code>##            
## Matched Sex 77 82 83 85 87 90
##       FALSE  0  0  2  1  1  0
##       TRUE   1  2  0  0  0  1</code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">mapped_read_count</span> <span class="op">~</span> <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">matching_sex</span>,
    ylab = <span class="st">"Mapped read count"</span>, xlab = <span class="st">"Matching sex"</span>
)
</pre></div>
<p><img src="recount-workshop_files/figure-html/matching%20sex-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>There are no discernible differences by case, age or mapped read count when compared with the matching sex status.</p>
</div>
<div id="adding-more-information" class="section level4">
<h4 class="hasAnchor">
<a href="#adding-more-information" class="anchor"></a>Adding more information</h4>
<p>Ultimately, more sample metadata information could be available elsewhere which can be useful for analyses. This information might be provided in the paper describing the data, the SRA Run Selector or other sources. As shown in Figure @ref(fig:Figure2), it is possible to append information to the <code>colData()</code> slot as long as there is a unique sample identifier such as the SRA Run id.</p>
<p>For our example use case, project SRP056604 has redundant information with the <code>characteristics</code> and the SRA Run selector <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP056604" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP056604</a>. However, this might not be the case for other projects. While the SRA Run Selector information is redundant in this case, we can practice adding it. We can download that information into text file named <code>SraRunTable.txt</code> by default, then load it into R, sort it appropriately and then append it to the <code>colData()</code> slot. Below we do so for the SRP056604 project.</p>
<div class="sourceCode" id="cb76"><pre class="downlit">
<span class="co">## Save the information from</span>
<span class="co">## https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP056604</span>
<span class="co">## to a table. We saved the file as SRP056604/SraRunTable.txt.</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"SraRunTable.txt"</span>))
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit">
<span class="co">## Read the table</span>
<span class="kw">sra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"SraRunTable.txt"</span>))

<span class="co">## Explore it</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">sra</span>)
</pre></div>
<pre><code>##          Run apoe_genotype Assay.Type AvgSpotLen       Bases  BioProject
## 1 SRR1931812           3/3    RNA-Seq         99 18563956983 PRJNA279526
## 2 SRR1931813           3/3    RNA-Seq         99 17851505463 PRJNA279526
## 3 SRR1931814           3/3    RNA-Seq         99 17673638103 PRJNA279526
## 4 SRR1931815           3/3    RNA-Seq         99 18671875299 PRJNA279526
## 5 SRR1931816           2/3    RNA-Seq         99 17533708731 PRJNA279526
## 6 SRR1931817 Not available    RNA-Seq         99 17215577181 PRJNA279526
##      BioSample braak_stage       Bytes Center.Name Consent DATASTORE.filetype
## 1 SAMN03448725          VI 12728839626         GEO  public          sra,fastq
## 2 SAMN03448726           V 12226156492         GEO  public          sra,fastq
## 3 SAMN03448730          VI 12122197205         GEO  public          fastq,sra
## 4 SAMN03448728          VI 13172185063         GEO  public          fastq,sra
## 5 SAMN03448727           I 12139059865         GEO  public          sra,fastq
## 6 SAMN03448729          II 11978968856         GEO  public          sra,fastq
##   DATASTORE.provider               DATASTORE.region Experiment gender
## 1         s3,ncbi,gs ncbi.public,s3.us-east-1,gs.US  SRX970047 female
## 2         s3,ncbi,gs ncbi.public,s3.us-east-1,gs.US  SRX970048 female
## 3         gs,s3,ncbi gs.US,s3.us-east-1,ncbi.public  SRX970049   male
## 4         ncbi,s3,gs s3.us-east-1,ncbi.public,gs.US  SRX970050 female
## 5         s3,ncbi,gs ncbi.public,gs.US,s3.us-east-1  SRX970051 female
## 6         gs,ncbi,s3 ncbi.public,s3.us-east-1,gs.US  SRX970052   male
##   GEO_Accession..exp.          Instrument LibraryLayout LibrarySelection
## 1          GSM1644996 Illumina HiSeq 2000        SINGLE             cDNA
## 2          GSM1644997 Illumina HiSeq 2000        SINGLE             cDNA
## 3          GSM1644998 Illumina HiSeq 2000        SINGLE             cDNA
## 4          GSM1644999 Illumina HiSeq 2000        SINGLE             cDNA
## 5          GSM1645000 Illumina HiSeq 2000        SINGLE             cDNA
## 6          GSM1645001 Illumina HiSeq 2000        SINGLE             cDNA
##    LibrarySource     Organism Platform          ReleaseDate Sample.Name
## 1 TRANSCRIPTOMIC Homo sapiens ILLUMINA 2015-04-01T00:00:00Z  GSM1644996
## 2 TRANSCRIPTOMIC Homo sapiens ILLUMINA 2015-04-01T00:00:00Z  GSM1644997
## 3 TRANSCRIPTOMIC Homo sapiens ILLUMINA 2015-04-01T00:00:00Z  GSM1644998
## 4 TRANSCRIPTOMIC Homo sapiens ILLUMINA 2015-04-01T00:00:00Z  GSM1644999
## 5 TRANSCRIPTOMIC Homo sapiens ILLUMINA 2015-04-01T00:00:00Z  GSM1645000
## 6 TRANSCRIPTOMIC Homo sapiens ILLUMINA 2015-04-01T00:00:00Z  GSM1645001
##                      source_name SRA.Study
## 1                LOAD_hippocampi SRP056604
## 2                LOAD_hippocampi SRP056604
## 3                LOAD_hippocampi SRP056604
## 4                LOAD_hippocampi SRP056604
## 5 age-matched control_hippocampi SRP056604
## 6 age-matched control_hippocampi SRP056604
##                              subject_group      tissue
## 1 LOAD (late onset of Alzheimer’s disease) Hippocampus
## 2 LOAD (late onset of Alzheimer’s disease) Hippocampus
## 3 LOAD (late onset of Alzheimer’s disease) Hippocampus
## 4 LOAD (late onset of Alzheimer’s disease) Hippocampus
## 5                      age-matched control Hippocampus
## 6                      age-matched control Hippocampus</code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit">
<span class="co">## Add the sra_ prefix to the variable names</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">sra</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"sra_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">sra</span>))

<span class="co">## Choose some variables we want to add</span>
<span class="kw">sra_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"sra_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"braak_stage"</span>, <span class="st">"gender"</span>, <span class="st">"tissue"</span>))

<span class="co">## Re-organize the SRA table based on the SRA Run ids we have</span>
<span class="kw">sra</span> <span class="op">&lt;-</span> <span class="kw">sra</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">run</span>, <span class="kw">sra</span><span class="op">$</span><span class="kw">sra_Run</span>), ]

<span class="co">## Double check the order</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">run</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">sra</span><span class="op">$</span><span class="kw">sra_Run</span>))
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb82"><pre class="downlit">
<span class="co">## Append the variables of interest</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>), <span class="kw">sra</span>[, <span class="kw">sra_vars</span>])

<span class="co">## Final dimensions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>## [1]  8 40</code></pre>
<div class="sourceCode" id="cb84"><pre class="downlit">
<span class="co">## Explore result</span>
<span class="fu">colData</span>(<span class="kw">rse_gene</span>)[, <span class="fl">38</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))]
</pre></div>
<pre><code>## DataFrame with 8 rows and 3 columns
##            sra_braak_stage  sra_gender  sra_tissue
##                &lt;character&gt; &lt;character&gt; &lt;character&gt;
## SRR1931819              II      female Hippocampus
## SRR1931818              II        male Hippocampus
## SRR1931817              II        male Hippocampus
## SRR1931816               I      female Hippocampus
## SRR1931815              VI      female Hippocampus
## SRR1931814              VI        male Hippocampus
## SRR1931813               V      female Hippocampus
## SRR1931812              VI      female Hippocampus</code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit">
<span class="co">## The sex information from the 'characteristics' and the</span>
<span class="co">## SRA Run Selector match</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sra_gender</span>, <span class="fu">colData</span>(<span class="kw">rse_gene</span>)<span class="op">$</span><span class="kw">sex</span>)
</pre></div>
<pre><code>##         
##          female male
##   female      5    0
##   male        0    3</code></pre>
</div>
<div id="add-curated-metadata-for-brain-studies" class="section level4">
<h4 class="hasAnchor">
<a href="#add-curated-metadata-for-brain-studies" class="anchor"></a>Add curated metadata for brain studies</h4>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure-1.png" alt="Example variables (blue column) and values of the data present in recount-brain." width="100%"><p class="caption">
Example variables (blue column) and values of the data present in recount-brain.
</p>
</div>
<p>To facilitate analyses of brain studies we further curated 62 SRA projects and merged them with GTEx and TCGA brain samples to create <code>recount-brain</code> <span class="citation">(16)</span> which contains brain sample metadata (Fig @ref(fig:Figure13)). SRP056604 is one of those 62 projects so we can easily include all this metadata to our RSE object using the <code><a href="https://rdrr.io/pkg/recount/man/add_metadata.html">add_metadata()</a></code> function in <code>recount</code>. It works very similar to <code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions()</a></code>.</p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/LieberInstitute/recountWorkshop2020/master/inst/vignettes/Figure2.jpg" alt="Uses of recount-brain and its relationship with recount2 . recount-brain facilitates identifying project(s) of interest (purple box) programmatically or interactively through https://jhubiostatistics.shinyapps.io/recount-brain/ . After downloading expression data from recount2 , recount-brain can enrich the sample metadata for brain studies. This information can be used to perform analyses to find differentially expressed genes and enriched gene sets such as those exemplified with SRA Study SRP027383 (Bao et al., 2014) , where the top differentially expressed gene among glioblastoma samples in recount2 is SCM4 . Black boxes represent R code with functions highlighted in blue, input arguments in green, and R objects in white." width="100%"><p class="caption">
Uses of recount-brain and its relationship with recount2 . recount-brain facilitates identifying project(s) of interest (purple box) programmatically or interactively through <a href="https://jhubiostatistics.shinyapps.io/recount-brain/" class="uri">https://jhubiostatistics.shinyapps.io/recount-brain/</a> . After downloading expression data from recount2 , recount-brain can enrich the sample metadata for brain studies. This information can be used to perform analyses to find differentially expressed genes and enriched gene sets such as those exemplified with SRA Study SRP027383 (Bao et al., 2014) , where the top differentially expressed gene among glioblastoma samples in recount2 is SCM4 . Black boxes represent R code with functions highlighted in blue, input arguments in green, and R objects in white.
</p>
</div>
<p>We can use <code><a href="https://rdrr.io/pkg/recount/man/add_predictions.html">add_predictions()</a></code> without specifying the <code>rse</code> argument as shown below to download the full <code>recount-brain</code> curated metadata (Fig @ref(fig:Figure14)).</p>
<div class="sourceCode" id="cb88"><pre class="downlit">
<span class="co">## Don't specify the rse argument to download the full recount_brain table</span>
<span class="kw">recount_brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/add_metadata.html">add_metadata</a></span>(source = <span class="st">"recount_brain_v2"</span>)
</pre></div>
<pre><code>## 2020-07-26 21:24:11 downloading the recount_brain metadata to /tmp/RtmpKDI1FD/recount_brain_v2.Rdata</code></pre>
<pre><code>## Loading objects:
##   recount_brain</code></pre>
<div class="sourceCode" id="cb91"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">recount_brain</span>)
</pre></div>
<pre><code>## [1] 6547   65</code></pre>
<div class="sourceCode" id="cb93"><pre class="downlit">
<span class="co">## Explore some of the variables for SRP056604</span>
<span class="kw">recount_brain</span>[<span class="kw">recount_brain</span><span class="op">$</span><span class="kw">Study_full</span> <span class="op">==</span> <span class="st">"SRP056604"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"age"</span>, <span class="st">"age_units"</span>,
    <span class="st">"sex"</span>, <span class="st">"rin"</span>, <span class="st">"pmi"</span>, <span class="st">"pmi_units"</span>, <span class="st">"pathology"</span>, <span class="st">"disease_status"</span>
)]
</pre></div>
<pre><code>##             age age_units    sex  rin  pmi pmi_units      pathology
## SRP056604.1  82     Years female 6.90 2.33     Hours Braak Stage VI
## SRP056604.2  87     Years female 6.50 2.00     Hours  Braak Stage V
## SRP056604.3  83     Years   male 7.00 2.83     Hours Braak Stage VI
## SRP056604.4  82     Years female 6.70 3.00     Hours Braak Stage VI
## SRP056604.5  77     Years female   NA 2.50     Hours  Braak Stage I
## SRP056604.6  83     Years   male 6.50 4.50     Hours Braak Stage II
## SRP056604.7  90     Years   male 8.60 2.00     Hours Braak Stage II
## SRP056604.8  85     Years female 8.32 1.00     Hours Braak Stage II
##             disease_status
## SRP056604.1        Disease
## SRP056604.2        Disease
## SRP056604.3        Disease
## SRP056604.4        Disease
## SRP056604.5        Control
## SRP056604.6        Control
## SRP056604.7        Control
## SRP056604.8        Control</code></pre>
<p>Or we can specify the <code>rse</code> argument and append these columns to our RSE object (Fig @ref(fig:Figure14)). In the case of the SRP056604 project, by using <code><a href="https://rdrr.io/pkg/recount/man/add_metadata.html">add_metadata()</a></code> we can skip over all the steps of downloading the SRA table and processing the GEO characteristics information, plus get additional information that was absent from the previous data sources such as the postmorterm interval (PMI) and the RNA integrity number (RIN).</p>
<div class="sourceCode" id="cb95"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>## [1]  8 40</code></pre>
<div class="sourceCode" id="cb97"><pre class="downlit">
<span class="kw">rse_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/add_metadata.html">add_metadata</a></span>(<span class="kw">rse_gene</span>, source = <span class="st">"recount_brain_v2"</span>)
</pre></div>
<pre><code>## 2020-07-26 21:24:12 downloading the recount_brain metadata to /tmp/RtmpKDI1FD/recount_brain_v2.Rdata</code></pre>
<pre><code>## Loading objects:
##   recount_brain</code></pre>
<pre><code>## 2020-07-26 21:24:13 found 8 out of 8 samples in the recount_brain metadata</code></pre>
<div class="sourceCode" id="cb101"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">colData</span>(<span class="kw">rse_gene</span>))
</pre></div>
<pre><code>## [1]   8 104</code></pre>
<p>Overall, you can explore interactively <code>recount-brain</code> at <a href="https://jhubiostatistics.shinyapps.io/recount-brain/">jhubiostatistics.shinyapps.io/recount-brain/</a>. Finally, note how <code><a href="https://rdrr.io/pkg/recount/man/add_metadata.html">add_metadata()</a></code> is designed to be expanded for more curated metadata, so if you would like to provide more data please let us know!</p>
</div>
</div>
<div id="de-setup" class="section level3">
<h3 class="hasAnchor">
<a href="#de-setup" class="anchor"></a>DE setup</h3>
<p>Now that we have all the metadata available we can perform a differential expression analysis. Lets look for differences by case when adjusting for the sex, age of the sample and PMI.</p>
<p>As we saw earlier in Figure @ref(fig:Figure9), it is important to scale the coverage counts to read counts. To highlight the fact that we scaled the counts, we will use a new object name and delete the previous one. However, in practice we would simply overwrite the <code>rse</code> object with the output of <code><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts(rse)</a></code>.</p>
<div class="sourceCode" id="cb103"><pre class="downlit">
<span class="co">## Scale counts</span>
<span class="kw">rse_gene_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="kw">rse_gene</span>)

<span class="co">## To highlight that we scaled the counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="kw">rse_gene</span>)
</pre></div>
</div>
<div id="de-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#de-analysis" class="anchor"></a>DE analysis</h3>
<p>Now that we have scaled the counts, there are multiple differential expression packages we could use as described elsewhere <span class="citation">(3,4)</span>. Since we have very few samples per group, we will use <code>DESeq2</code> <span class="citation">(18)</span>. The model we will use will test for differential expression between LOAD and control samples, adjusting for sex and age. In a real use case we might have to explore the results with different models.</p>
<div class="sourceCode" id="cb104"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/mikelove/DESeq2">"DESeq2"</a></span>)

<span class="co">## Specify design and switch to DESeq2 format</span>
<span class="kw">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="kw">rse_gene_scaled</span>, <span class="op">~</span> <span class="kw">sex</span> <span class="op">+</span> <span class="kw">age</span> <span class="op">+</span> <span class="kw">pmi</span> <span class="op">+</span> <span class="kw">case</span>)
</pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>##   the design formula contains one or more numeric variables with integer values,
##   specifying a model with increasing fold change for higher values.
##   did you mean for this to be a factor? if so, first convert
##   this variable to a factor using the factor() function</code></pre>
<pre><code>##   the design formula contains one or more numeric variables that have mean or
##   standard deviation larger than 5 (an arbitrary threshold to trigger this message).
##   it is generally a good idea to center and scale numeric variables in the design
##   to improve GLM convergence.</code></pre>
<div class="sourceCode" id="cb108"><pre class="downlit">
<span class="co">## Perform DE analysis</span>
<span class="kw">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="kw">dds</span>,
    test = <span class="st">"LRT"</span>, reduced = <span class="op">~</span> <span class="kw">sex</span> <span class="op">+</span> <span class="kw">age</span> <span class="op">+</span> <span class="kw">pmi</span>,
    fitType = <span class="st">"local"</span>
)
</pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## 2 rows did not converge in beta, labelled in mcols(object)$fullBetaConv. Use larger maxit argument with nbinomLRT</code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit">
<span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="kw">dds</span>,
    main = <span class="st">"DESeq2 results for SRP056604"</span>,
    contrast = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"case"</span>, <span class="st">"LOAD"</span>, <span class="st">"control"</span>), alpha = <span class="fl">0.01</span>
)
</pre></div>
<p><img src="recount-workshop_files/figure-html/deseq2_de-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>We can use <code>ideal</code> <span class="citation">(19)</span> to make a volcano plot of the DE results.</p>
<div class="sourceCode" id="cb117"><pre class="downlit">
<span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="kw">dds</span>, contrast = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"case"</span>, <span class="st">"LOAD"</span>, <span class="st">"control"</span>), alpha = <span class="fl">0.01</span>)

<span class="co">## Make a volcano plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/federicomarini/ideal">"ideal"</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/ideal/man/plot_volcano.html">plot_volcano</a></span>(<span class="kw">res</span>, FDR = <span class="fl">0.01</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/ideal%20volcano-1.png" width="700"></p>
<p>Having run the DE analysis, we can explore some of the top results either with an MA plot and a volcano plot. Both reveal strong differential expression signal for several genes.</p>
</div>
<div id="de-report" class="section level3">
<h3 class="hasAnchor">
<a href="#de-report" class="anchor"></a>DE report</h3>
<p>Now that we have the differential expression results, we can use some of the tools with the biocView <a href="http://bioconductor.org/packages/release/BiocViews.html#___ReportWriting">ReportWriting</a> to create a report. One of them is <code>regionReport</code> <span class="citation">(20)</span> which can create reports from <code>DESeq2</code> <span class="citation">(18)</span> and <code>edgeR</code> <span class="citation">(21)</span> results. It can also handle <code>limma-voom</code> <span class="citation">(22)</span> results by making them look like <code>DESeq2</code> results.</p>
<p>We can now create the report, which should open automatically in a browser.</p>
<div class="sourceCode" id="cb118"><pre class="downlit">
<span class="co">## Make a report with the results</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/regionReport">"regionReport"</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="kw">dds</span>,
    res = <span class="kw">res</span>, project = <span class="st">"SRP056604"</span>,
    intgroup = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"sex"</span>, <span class="st">"case"</span>), outdir = <span class="st">"."</span>,
    output = <span class="st">"SRP056604_main-results"</span>
)
</pre></div>
<p>If the report does not open automatically, we can open it with <code><a href="https://rdrr.io/r/utils/browseURL.html">browseURL()</a></code>.</p>
<div class="sourceCode" id="cb119"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html">browseURL</a></span>(<span class="st">"SRP056604_main-results.html"</span>)
</pre></div>
</div>
<div id="go-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#go-enrichment" class="anchor"></a>GO enrichment</h3>
<p>Using <code>clusterProfiler</code> <span class="citation">(23)</span> we can then perform several enrichment analyses. The gene names from the recount2 objects use Gencode ids, which do not work by default with <code>org.Hs.eg.db</code>. To make sure they do, we can simply change the ids to Ensembl ids. Here we show how to perform an enrichment analysis using the biological process ontology. We will the genes that have a p-value as the universe background.</p>
<div class="sourceCode" id="cb120"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://guangchuangyu.github.io/software/clusterProfiler">"clusterProfiler"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"org.Hs.eg.db"</span>)

<span class="co">## Remember that dds had ENSEMBL ids for the genes</span>
<span class="kw">ensembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">dds</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">ensembl</span>)
</pre></div>
<pre><code>## [1] "ENSG00000000003" "ENSG00000000005" "ENSG00000000419" "ENSG00000000457"
## [5] "ENSG00000000460" "ENSG00000000938"</code></pre>
<div class="sourceCode" id="cb122"><pre class="downlit">
<span class="co">## Not all genes have a p-value</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">res</span><span class="op">$</span><span class="kw">padj</span>))
</pre></div>
<pre><code>## 
## FALSE  TRUE 
## 15799 42238</code></pre>
<div class="sourceCode" id="cb124"><pre class="downlit">
<span class="co">## Perform enrichment analysis for Biological Process (BP)</span>
<span class="co">## Note that the argument is keytype instead of keyType in Bioconductor 3.5</span>
<span class="kw">enrich_go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span>(
    gene = <span class="kw">ensembl</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>)],
    OrgDb = <span class="kw">org.Hs.eg.db</span>, keyType = <span class="st">"ENSEMBL"</span>, ont = <span class="st">"BP"</span>,
    pAdjustMethod = <span class="st">"BH"</span>, pvalueCutoff = <span class="fl">0.01</span>, qvalueCutoff = <span class="fl">0.05</span>,
    universe = <span class="kw">ensembl</span>[<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">res</span><span class="op">$</span><span class="kw">padj</span>)]
)

<span class="co">## Visualize enrichment results</span>
<span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span>(<span class="kw">enrich_go</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/go_analysis-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>Several other analyses can be performed with the resulting list of differentially expressed genes as described previously <span class="citation">(3,4)</span>, although that is beyond the scope of this workshop.</p>
</div>
<div id="secondary-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#secondary-analysis" class="anchor"></a>Secondary analysis</h3>
<p>Since we noticed that the sex predictions and the reported sex do not match for half of the samples, we can check if there are any genes associated with whether the sex matched.</p>
<div class="sourceCode" id="cb125"><pre class="downlit">
<span class="co">## DE analysis checking what genes are different by matching sex</span>
<span class="kw">dds2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="kw">rse_gene_scaled</span>, <span class="op">~</span><span class="kw">matching_sex</span>)
</pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit">
<span class="kw">dds2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="kw">dds2</span>, test = <span class="st">"LRT"</span>, reduced = <span class="op">~</span><span class="fl">1</span>, fitType = <span class="st">"local"</span>)
</pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit">
<span class="kw">res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="kw">dds2</span>, alpha = <span class="fl">0.01</span>)

<span class="co">## Visually inspect results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="kw">res2</span>, main = <span class="st">"DESeq2 results for SRP056604 - sex predictions"</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/secondary%20analysis-1.png" width="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb135"><pre class="downlit">
<span class="co">## Lets add gene symbols to the volcano plot, they are stored in</span>
<span class="co">## the rowRanges slot of the rse object</span>
<span class="fu">rowRanges</span>(<span class="kw">rse_gene_scaled</span>)
</pre></div>
<pre><code>## GRanges object with 58037 ranges and 3 metadata columns:
##                      seqnames              ranges strand |            gene_id
##                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;
##   ENSG00000000003.14     chrX 100627109-100639991      - | ENSG00000000003.14
##    ENSG00000000005.5     chrX 100584802-100599885      + |  ENSG00000000005.5
##   ENSG00000000419.12    chr20   50934867-50958555      - | ENSG00000000419.12
##   ENSG00000000457.13     chr1 169849631-169894267      - | ENSG00000000457.13
##   ENSG00000000460.16     chr1 169662007-169854080      + | ENSG00000000460.16
##                  ...      ...                 ...    ... .                ...
##    ENSG00000283695.1    chr19   52865369-52865429      - |  ENSG00000283695.1
##    ENSG00000283696.1     chr1 161399409-161422424      + |  ENSG00000283696.1
##    ENSG00000283697.1     chrX 149548210-149549852      - |  ENSG00000283697.1
##    ENSG00000283698.1     chr2 112439312-112469687      - |  ENSG00000283698.1
##    ENSG00000283699.1    chr10   12653138-12653197      - |  ENSG00000283699.1
##                      bp_length          symbol
##                      &lt;integer&gt; &lt;CharacterList&gt;
##   ENSG00000000003.14      4535          TSPAN6
##    ENSG00000000005.5      1610            TNMD
##   ENSG00000000419.12      1207            DPM1
##   ENSG00000000457.13      6883           SCYL3
##   ENSG00000000460.16      5967        C1orf112
##                  ...       ...             ...
##    ENSG00000283695.1        61              NA
##    ENSG00000283696.1       997              NA
##    ENSG00000283697.1      1184           HSFX3
##    ENSG00000283698.1       940              NA
##    ENSG00000283699.1        60         MIR4481
##   -------
##   seqinfo: 25 sequences (1 circular) from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb137"><pre class="downlit">
<span class="kw">res2</span><span class="op">$</span><span class="kw">symbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">rowRanges</span>(<span class="kw">rse_gene_scaled</span>)<span class="op">$</span><span class="kw">symbol</span>, <span class="st">"[["</span>, <span class="fl">1</span>)

<span class="co">## Select some DE genes</span>
<span class="kw">intgenes</span> <span class="op">&lt;-</span> <span class="kw">res2</span><span class="op">$</span><span class="kw">symbol</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res2</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.0005</span>)]
<span class="kw">intgenes</span> <span class="op">&lt;-</span> <span class="kw">intgenes</span>[<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">intgenes</span>)]

<span class="co">## Make the volcano plot</span>
<span class="fu"><a href="https://rdrr.io/pkg/ideal/man/plot_volcano.html">plot_volcano</a></span>(<span class="kw">res2</span>, FDR = <span class="fl">0.01</span>, intgenes = <span class="kw">intgenes</span>)
</pre></div>
<pre><code>## Warning: Removed 44 rows containing missing values (geom_point).</code></pre>
<p><img src="recount-workshop_files/figure-html/secondary%20analysis-2.png" width="100%" style="display: block; margin: auto;"></p>
<p>We can create a report just like before.</p>
<div class="sourceCode" id="cb139"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="kw">dds2</span>,
    res = <span class="kw">res2</span>, project = <span class="st">"SRP056604 - matching sex"</span>,
    intgroup = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"sex"</span>, <span class="st">"predicted_sex"</span>), outdir = <span class="st">"."</span>,
    output = <span class="st">"SRP056604_sex-predictions"</span>
)
</pre></div>
<p>Checking the chromosome of the DE genes by matching sex status shows that most of these genes are not in chromosomes X and Y as shown below.</p>
<div class="sourceCode" id="cb140"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">seqnames</span>(<span class="fu">rowRanges</span>(<span class="kw">rse_gene_scaled</span>))[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res2</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.01</span>)]))
</pre></div>
<pre><code>## 
## chr13 chr14 chr16 chr18 chr21  chrY  chrM  chr1  chr5  chr9 chr10  chr2  chr4 
##     0     0     0     0     0     0     0     1     1     1     1     2     2 
##  chr7 chr22  chrX  chr3  chr6 chr12 chr20  chr8 chr15 chr19 chr11 chr17 
##     2     2     2     3     3     3     3     4     4     4     5     5</code></pre>
</div>
</div>
<div id="other-features" class="section level2">
<h2 class="hasAnchor">
<a href="#other-features" class="anchor"></a>Other features</h2>
<p>As described in Figure @ref(fig:Figure1), recount2 provides data for expression features beyond genes. In this section we perform a differential expression analysis using the exon data as well as the base-pair resolution information.</p>
<div id="exon-and-exon-exon-junctions" class="section level3">
<h3 class="hasAnchor">
<a href="#exon-and-exon-exon-junctions" class="anchor"></a>Exon and exon-exon junctions</h3>
<p>The exon and exon-exon junction coverage count matrices are similar to the gene level one and can also be downloaded with <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code>. However, these coverage count matrices are much larger than the gene one. Aggressive filtering of lowly expressed exons or exon-exon junctions can reduce the matrix dimensions if this impacts the performance of the differential expression software used.</p>
<p>Below we repeat the gene level analysis for the disjoint exon data. We first download the exon data, add the expanded metadata we constructed for the gene analysis, and then perform the differential expression analysis using <code>limma-voom</code>.</p>
<div class="sourceCode" id="cb142"><pre class="downlit">
<span class="co">## Download the data if it is not there</span>
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"rse_exon.Rdata"</span>))) {
    <span class="co">## In case you decide to download the data instead of using the</span>
    <span class="co">## pre-installed data</span>
    <span class="kw">local_path</span> <span class="op">&lt;-</span> <span class="st">"SRP056604"</span>
    <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span>(<span class="st">"SRP056604"</span>, type = <span class="st">"rse-exon"</span>)
}

<span class="co">## Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"rse_exon.Rdata"</span>))

<span class="co">## Scale and add the metadata (it is in the same order)</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu">colData</span>(<span class="kw">rse_exon</span>)<span class="op">$</span><span class="kw">run</span>, <span class="fu">colData</span>(<span class="kw">rse_gene_scaled</span>)<span class="op">$</span><span class="kw">run</span>)
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb144"><pre class="downlit">
<span class="fu">colData</span>(<span class="kw">rse_exon</span>) <span class="op">&lt;-</span> <span class="fu">colData</span>(<span class="kw">rse_gene_scaled</span>)
<span class="kw">rse_exon_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span>(<span class="kw">rse_exon</span>)
<span class="co">## To highlight that we scaled the counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="kw">rse_exon</span>)

<span class="co">## Filter lowly expressed exons: reduces the object size</span>
<span class="co">## and run time</span>
<span class="kw">filter_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="fu">assays</span>(<span class="kw">rse_exon_scaled</span>)<span class="op">$</span><span class="kw">counts</span>) <span class="op">&gt;</span> <span class="fl">5</span>
<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="kw">filter_exon</span>) <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">filter_exon</span>) <span class="op">*</span> <span class="fl">100</span>, <span class="fl">2</span>)
</pre></div>
<pre><code>## filter_exon
## FALSE  TRUE 
## 62.88 37.12</code></pre>
<div class="sourceCode" id="cb146"><pre class="downlit">
<span class="co">## Perform the filtering and change default names</span>
<span class="kw">rse_e</span> <span class="op">&lt;-</span> <span class="kw">rse_exon_scaled</span>[<span class="kw">filter_exon</span>, ]
<span class="fu">rowRanges</span>(<span class="kw">rse_e</span>)<span class="op">$</span><span class="kw">gene_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">rse_e</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">rse_e</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"exon_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">rse_e</span>)))

<span class="co">## Create DESeq2 object for the exon data</span>
<span class="kw">dds_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="kw">rse_e</span>, <span class="op">~</span> <span class="kw">sex</span> <span class="op">+</span> <span class="kw">age</span> <span class="op">+</span> <span class="kw">pmi</span> <span class="op">+</span> <span class="kw">case</span>)
</pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>##   the design formula contains one or more numeric variables with integer values,
##   specifying a model with increasing fold change for higher values.
##   did you mean for this to be a factor? if so, first convert
##   this variable to a factor using the factor() function</code></pre>
<pre><code>##   the design formula contains one or more numeric variables that have mean or
##   standard deviation larger than 5 (an arbitrary threshold to trigger this message).
##   it is generally a good idea to center and scale numeric variables in the design
##   to improve GLM convergence.</code></pre>
<div class="sourceCode" id="cb150"><pre class="downlit">
<span class="co">## Perform DE analysis</span>
<span class="kw">dds_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="kw">dds_exon</span>,
    test = <span class="st">"LRT"</span>, reduced = <span class="op">~</span> <span class="kw">sex</span> <span class="op">+</span> <span class="kw">age</span> <span class="op">+</span> <span class="kw">pmi</span>,
    fitType = <span class="st">"local"</span>
)
</pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb157"><pre class="downlit">
<span class="kw">res_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="kw">dds_exon</span>,
    contrast = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"case"</span>, <span class="st">"LOAD"</span>, <span class="st">"control"</span>),
    alpha = <span class="fl">0.01</span>
)

<span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="kw">dds_exon</span>,
    main = <span class="st">"DESeq2 results for SRP056604 -exon level"</span>,
    contrast = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"case"</span>, <span class="st">"LOAD"</span>, <span class="st">"control"</span>), alpha = <span class="fl">0.01</span>
)
</pre></div>
<p><img src="recount-workshop_files/figure-html/exon_de_analysis-1.png" width="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb158"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/ideal/man/plot_volcano.html">plot_volcano</a></span>(<span class="kw">res_exon</span>, FDR = <span class="fl">0.01</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/exon_de_analysis-2.png" width="100%" style="display: block; margin: auto;"></p>
<p>Just like at the gene level, we see several exons differentially expressed between LOAD and controls samples. As a first step to integrate the results from the two features, we can compare the list of genes that are differentially expressed versus the genes that have at least one exon differentially expressed.</p>
<div class="sourceCode" id="cb159"><pre class="downlit">
<span class="co">## Get the gene ids for genes that are DE at the gene level or that have at</span>
<span class="co">## least one exon with DE signal.</span>
<span class="kw">genes_w_de_exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu">rowRanges</span>(<span class="kw">rse_e</span>)<span class="op">$</span><span class="kw">gene_id</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res_exon</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.01</span>)])
<span class="kw">genes_de</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">rse_gene_scaled</span>)[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.01</span>)]

<span class="co">## Make a venn diagram</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/talgalili/gplots">"gplots"</a></span>)
<span class="kw">vinfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gplots/man/venn.html">venn</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"genes"</span> = <span class="kw">genes_de</span>, <span class="st">"exons"</span> = <span class="kw">genes_w_de_exon</span>),
    names = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"genes"</span>, <span class="st">"exons"</span>), show.plot = <span class="fl">FALSE</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="kw">vinfo</span>) <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Genes with DE signal: at the gene and exon levels"</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/gene_exon-1.png" width="100%" style="display: block; margin: auto;"></p>
<pre><code>## integer(0)</code></pre>
<p>Not all differentially expressed genes have differentially expressed exons, nor genes with at least one differentially expressed exon are necessarily differentially expressed. This is in line with what was described in Figure 2B of Soneson et al., 2015 <span class="citation">(24)</span>.</p>
<p>This was just a quick example of how we can perform differential expression analyses at the gene and exon feature levels. We envision that more involved pipelines could be developed that leverage both feature levels such as in Jaffe at al., 2018 <span class="citation">(25)</span>. For instance, we could focus on the differentially expressed genes with at least one differentially expressed exon and compare the direction of the DE signal versus the gene level signal as shown below.</p>
<div class="sourceCode" id="cb161"><pre class="downlit">
<span class="co">## Keep only the DE exons that are from a gene that is also DE</span>
<span class="kw">top_exon_de</span> <span class="op">&lt;-</span> <span class="kw">res_exon</span>[<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">intersect</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res_exon</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.01</span>),
    <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu">rowRanges</span>(<span class="kw">rse_e</span>)<span class="op">$</span><span class="kw">gene_id</span> <span class="op">%in%</span>
        <span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span>(<span class="kw">vinfo</span>, <span class="st">"intersections"</span>)[[<span class="st">"genes:exons"</span>]])
), ]
<span class="co">## Add the gene id</span>
<span class="kw">top_exon_de</span><span class="op">$</span><span class="kw">gene_id</span> <span class="op">&lt;-</span> <span class="fu">rowRanges</span>(<span class="kw">rse_e</span>)<span class="op">$</span><span class="kw">gene_id</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">top_exon_de</span>),
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">rse_e</span>)
)]

<span class="co">## Find the fold change that is the most extreme among the DE exons of a gene</span>
<span class="kw">exon_max_fc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html">tapply</a></span>(
    <span class="kw">top_exon_de</span><span class="op">$</span><span class="kw">log2FoldChange</span>, <span class="kw">top_exon_de</span><span class="op">$</span><span class="kw">gene_id</span>,
    <span class="fu">function</span>(<span class="kw">x</span>) {
        <span class="kw">x</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="kw">x</span>))]
    }
)

<span class="co">## Keep only the DE genes that match the previous selection</span>
<span class="kw">top_gene_de</span> <span class="op">&lt;-</span> <span class="kw">res</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">exon_max_fc</span>), <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">res</span>)), ]

<span class="co">## Make the plot</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="kw">top_gene_de</span><span class="op">$</span><span class="kw">log2FoldChange</span>, <span class="kw">exon_max_fc</span>,
    pch = <span class="fl">20</span>,
    col = <span class="fu"><a href="https://rdrr.io/r/grDevices/adjustcolor.html">adjustcolor</a></span>(<span class="st">"black"</span>, <span class="fl">1</span> <span class="op">/</span> <span class="fl">2</span>),
    ylab = <span class="st">"Most extreme log FC at the exon level among DE exons"</span>,
    xlab = <span class="st">"Log fold change (FC) at the gene level"</span>,
    main = <span class="st">"DE genes with at least one DE exon"</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(a = <span class="fl">0</span>, b = <span class="fl">1</span>, col = <span class="st">"red"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(h = <span class="fl">0</span>, col = <span class="st">"grey80"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(v = <span class="fl">0</span>, col = <span class="st">"grey80"</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/gene_exon_match-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>The fold change for most exons shown above agrees with the gene level fold change. In data from other projects, some of the fold changes have opposite directions and could be interesting to study further.</p>
</div>
<div id="base-pair-resolution" class="section level3">
<h3 class="hasAnchor">
<a href="#base-pair-resolution" class="anchor"></a>Base-pair resolution</h3>
<p>recount2 provides BigWig coverage files (unscaled) for all samples as well as a mean BigWig coverage file per project where each sample was scaled to 40 million 100 base-pair reads. The mean BigWig files are exactly what is needed to start an <em>expressed regions</em> analysis with <code>derfinder</code> <span class="citation">(9)</span>. <code>recount</code> provides two related functions: <code><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html">expressed_regions()</a></code> which is used to define a set of regions based on the mean BigWig file for a given project, and <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code> which based on a set of regions builds a count coverage matrix in a <em>RangedSummarizedExperiment</em> object just like the ones that are provided for genes and exons. <code><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study()</a></code> with <code>type = "mean"</code> or <code>type = "samples"</code> can be used to download the BigWig files, which we recommend doing when working with them extensively.</p>
<p>For illustrative purposes, we will use the data from chromosome 12 for the SRP056604 project. We chose chromosome 12 based on the number of DE genes per chromosome</p>
<div class="sourceCode" id="cb162"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">seqnames</span>(<span class="fu">rowRanges</span>(<span class="kw">rse_gene_scaled</span>)[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">res</span><span class="op">$</span><span class="kw">padj</span> <span class="op">&lt;</span> <span class="fl">0.01</span>)])),
    decreasing = <span class="fl">TRUE</span>
)
</pre></div>
<pre><code>## 
##  chr1  chr7 chr12  chrM  chr6 chr15  chr2 chr17 chr11 chr14 chr19  chr4  chr5 
##    10     8     7     7     6     6     5     5     4     4     4     3     3 
##  chr9 chr10 chr16 chr21 chr22  chr3  chr8 chr13 chr18  chrX chr20  chrY 
##     3     3     3     3     3     2     2     2     2     2     1     0</code></pre>
<p>First, we obtain the expressed regions using a relatively high mean cutoff of 5. We then filter the regions to keep only the ones longer than 100 base-pairs to shorten the time needed for running <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code>.</p>
<div class="sourceCode" id="cb164"><pre class="downlit">
<span class="co">## Define expressed regions for study SRP056604, only for chromosome 12</span>
<span class="kw">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/expressed_regions.html">expressed_regions</a></span>(<span class="st">"SRP056604"</span>, <span class="st">"chr12"</span>,
    cutoff = <span class="fl">5L</span>,
    maxClusterGap = <span class="fl">3000L</span>, outdir = <span class="kw">local_path</span>
)
</pre></div>
<pre><code>## 2020-07-26 21:28:23 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/mean_SRP056604.bw</code></pre>
<pre><code>## 2020-07-26 21:28:33 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-07-26 21:29:06 filterData: originally there were 133275309 rows, now there are 133275309 rows. Meaning that 0 percent was filtered.</code></pre>
<pre><code>## 2020-07-26 21:29:06 findRegions: identifying potential segments</code></pre>
<pre><code>## 2020-07-26 21:29:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-07-26 21:29:06 .getSegmentsRle: segmenting with cutoff(s) 5</code></pre>
<pre><code>## 2020-07-26 21:29:14 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-07-26 21:29:14 findRegions: identifying region clusters</code></pre>
<div class="sourceCode" id="cb173"><pre class="downlit">
<span class="co">## Explore the resulting expressed regions</span>
<span class="kw">regions</span>
</pre></div>
<pre><code>## GRanges object with 45415 ranges and 6 metadata columns:
##         seqnames              ranges strand |     value      area indexStart
##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##       1    chr12         14557-14609      * |   5.51229   292.151      14557
##       2    chr12         14694-14944      * |  18.51754  4647.903      14694
##       3    chr12         15085-15153      * |  32.19300  2221.317      15085
##       4    chr12         15489-15589      * |  15.51833  1567.352      15489
##       5    chr12         15889-16065      * |  51.57212  9128.266      15889
##     ...      ...                 ...    ... .       ...       ...        ...
##   45411    chr12 133204363-133204806      * |  14.71408 6533.0499  133204363
##   45412    chr12 133205808-133205818      * |   5.28276   58.1103  133205808
##   45413    chr12 133206109-133206245      * |   9.16745 1255.9413  133206109
##   45414    chr12 133206247-133206273      * |   5.30815  143.3200  133206247
##   45415    chr12 133206625-133206646      * |   5.22781  115.0117  133206625
##          indexEnd cluster clusterL
##         &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
##       1     14609       1    16374
##       2     14944       1    16374
##       3     15153       1    16374
##       4     15589       1    16374
##       5     16065       1    16374
##     ...       ...     ...      ...
##   45411 133204806    4711     4503
##   45412 133205818    4711     4503
##   45413 133206245    4711     4503
##   45414 133206273    4711     4503
##   45415 133206646    4711     4503
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<div class="sourceCode" id="cb175"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="fu">width</span>(<span class="kw">regions</span>))
</pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##     1.0     9.0    61.0    88.3   120.0  5483.0</code></pre>
<div class="sourceCode" id="cb177"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">width</span>(<span class="kw">regions</span>) <span class="op">&gt;=</span> <span class="fl">60</span>)
</pre></div>
<pre><code>## 
## FALSE  TRUE 
## 22465 22950</code></pre>
<div class="sourceCode" id="cb179"><pre class="downlit">
<span class="co">## Keep only the ones that are at least 60 bp long</span>
<span class="kw">regions</span> <span class="op">&lt;-</span> <span class="kw">regions</span>[<span class="fu">width</span>(<span class="kw">regions</span>) <span class="op">&gt;=</span> <span class="fl">60</span>]
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">regions</span>)
</pre></div>
<pre><code>## [1] 22950</code></pre>
<p>Now that we have a set of regions to work with, we proceed to build a <em>RangedSummarizedExperiment</em> object with the coverage counts, add the expanded metadata we built for the gene level, and scale the counts. Note that <code><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix()</a></code> by defaults scales the counts. We will round them to integers in order to use <code>DESeq2</code>.</p>
<div class="sourceCode" id="cb181"><pre class="downlit">
<span class="co">## Compute coverage matrix for study SRP056604, only for chromosome 12</span>
<span class="co">## Takes about 45 seconds with local data</span>
<span class="co">## and about 70 seconds with data from the web (outdir = NULL)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="kw">rse_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/coverage_matrix.html">coverage_matrix</a></span>(<span class="st">"SRP056604"</span>, <span class="st">"chr12"</span>, <span class="kw">regions</span>,
    chunksize = <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">regions</span>), outdir = <span class="kw">local_path</span>, round = <span class="fl">TRUE</span>
))
</pre></div>
<pre><code>## 2020-07-26 21:29:21 downloading file SRP056604.tsv to /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604</code></pre>
<pre><code>## 2020-07-26 21:29:22 railMatrix: processing regions 1 to 22950</code></pre>
<pre><code>## 2020-07-26 21:29:22 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931819.bw</code></pre>
<pre><code>## 2020-07-26 21:29:25 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931818.bw</code></pre>
<pre><code>## 2020-07-26 21:29:28 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931817.bw</code></pre>
<pre><code>## 2020-07-26 21:29:31 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931816.bw</code></pre>
<pre><code>## 2020-07-26 21:29:33 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931815.bw</code></pre>
<pre><code>## 2020-07-26 21:29:36 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931814.bw</code></pre>
<pre><code>## 2020-07-26 21:29:39 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931813.bw</code></pre>
<pre><code>## 2020-07-26 21:29:42 railMatrix: processing file /usr/local/lib/R/host-site-library/recountWorkshop2020/extdata/SRP056604/bw/SRR1931812.bw</code></pre>
<pre><code>##    user  system elapsed 
##  21.710   1.415  23.841</code></pre>
<div class="sourceCode" id="cb193"><pre class="downlit">
<span class="co">## Use the expanded metadata we built for the gene model</span>
<span class="fu">colData</span>(<span class="kw">rse_er</span>) <span class="op">&lt;-</span> <span class="fu">colData</span>(<span class="kw">rse_gene_scaled</span>)
</pre></div>
<p>Now that we have an integer count matrix for the expressed regions, we can proceed with the differential expression analysis just like we did at the gene and exon feature levels.</p>
<div class="sourceCode" id="cb194"><pre class="downlit">
<span class="co">## Define DESeq2 object</span>
<span class="kw">dds_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(<span class="kw">rse_er</span>, <span class="op">~</span> <span class="kw">sex</span> <span class="op">+</span> <span class="kw">age</span> <span class="op">+</span> <span class="kw">pmi</span> <span class="op">+</span> <span class="kw">case</span>)
</pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>##   the design formula contains one or more numeric variables with integer values,
##   specifying a model with increasing fold change for higher values.
##   did you mean for this to be a factor? if so, first convert
##   this variable to a factor using the factor() function</code></pre>
<pre><code>##   the design formula contains one or more numeric variables that have mean or
##   standard deviation larger than 5 (an arbitrary threshold to trigger this message).
##   it is generally a good idea to center and scale numeric variables in the design
##   to improve GLM convergence.</code></pre>
<div class="sourceCode" id="cb198"><pre class="downlit">
<span class="kw">dds_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="kw">dds_er</span>,
    test = <span class="st">"LRT"</span>, reduced = <span class="op">~</span> <span class="kw">sex</span> <span class="op">+</span> <span class="kw">age</span> <span class="op">+</span> <span class="kw">pmi</span>,
    fitType = <span class="st">"local"</span>
)
</pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb205"><pre class="downlit">
<span class="kw">res_er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="kw">dds_er</span>, alpha = <span class="fl">0.05</span>, contrast = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"case"</span>, <span class="st">"LOAD"</span>, <span class="st">"control"</span>))

<span class="co">## Visually inspect results</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/plotMA.html">plotMA</a></span>(<span class="kw">res_er</span>,
    main = <span class="st">"DESeq2 results for SRP056604 - DERs"</span>,
    contrast = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"case"</span>, <span class="st">"LOAD"</span>, <span class="st">"control"</span>), alpha = <span class="fl">0.05</span>
)
</pre></div>
<pre><code>## Warning in plot.window(...): "contrast" is not a graphical parameter</code></pre>
<pre><code>## Warning in plot.xy(xy, type, ...): "contrast" is not a graphical parameter</code></pre>
<pre><code>## Warning in axis(side = side, at = at, labels = labels, ...): "contrast" is not a
## graphical parameter

## Warning in axis(side = side, at = at, labels = labels, ...): "contrast" is not a
## graphical parameter</code></pre>
<pre><code>## Warning in box(...): "contrast" is not a graphical parameter</code></pre>
<pre><code>## Warning in title(...): "contrast" is not a graphical parameter</code></pre>
<p><img src="recount-workshop_files/figure-html/er_de_analysis-1.png" width="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb211"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/ideal/man/plot_volcano.html">plot_volcano</a></span>(<span class="kw">res_er</span>, FDR = <span class="fl">0.05</span>)
</pre></div>
<p><img src="recount-workshop_files/figure-html/er_de_analysis-2.png" width="100%" style="display: block; margin: auto;"></p>
<p>We can also create a report, just like before.</p>
<div class="sourceCode" id="cb212"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/regionReport/man/DESeq2Report.html">DESeq2Report</a></span>(<span class="kw">dds_er</span>,
    res = <span class="kw">res_er</span>, project = <span class="st">"SRP056604 - DERs"</span>,
    intgroup = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"sex"</span>, <span class="st">"case"</span>), outdir = <span class="st">"."</span>,
    output = <span class="st">"SRP056604_DERs"</span>
)
</pre></div>
<p>Having identified the differentially expressed regions (DERs), we can sort all regions by their adjusted p-value.</p>
<div class="sourceCode" id="cb213"><pre class="downlit">
<span class="co">## Sort regions by q-value</span>
<span class="kw">regions_by_padj</span> <span class="op">&lt;-</span> <span class="kw">regions</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">res_er</span><span class="op">$</span><span class="kw">padj</span>, decreasing = <span class="fl">FALSE</span>)]

<span class="co">## Look at the top 10</span>
<span class="kw">regions_by_padj</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>]
</pre></div>
<pre><code>## GRanges object with 10 ranges and 6 metadata columns:
##         seqnames              ranges strand |     value      area indexStart
##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##   19184    chr12   56652770-56653028      * |   7.86537  2037.132   56652770
##   19183    chr12   56646007-56646220      * |  49.93447 10685.977   56646007
##   19191    chr12   56660076-56660174      * |   6.98547   691.562   56660076
##    3971    chr12     6968608-6968702      * |   6.20458   589.435    6968608
##   19211    chr12   56666626-56666848      * |  10.06840  2245.253   56666626
##   19226    chr12   56670429-56670523      * |   7.24983   688.734   56670429
##    3545    chr12     6581473-6581591      * | 104.96965 12491.389    6581473
##   19189    chr12   56657528-56657606      * |   5.61127   443.290   56657528
##   39548    chr12 120645698-120646104      * |  24.13298  9822.124  120645698
##   19200    chr12   56662428-56662558      * |   6.35010   831.862   56662428
##          indexEnd cluster clusterL
##         &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
##   19184  56653028    1996    36189
##   19183  56646220    1995    10725
##   19191  56660174    1996    36189
##    3971   6968702     309     5639
##   19211  56666848    1996    36189
##   19226  56670523    1996    36189
##    3545   6581591     286    86353
##   19189  56657606    1996    36189
##   39548 120646104    4162    26897
##   19200  56662558    1996    36189
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<div class="sourceCode" id="cb215"><pre class="downlit">
<span class="fu">width</span>(<span class="kw">regions_by_padj</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>])
</pre></div>
<pre><code>##  [1] 259 214  99  95 223  95 119  79 407 131</code></pre>
</div>
<div id="visualize-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-regions" class="anchor"></a>Visualize regions</h3>
<p>Since the DERs do not necessarily match the annotation, it is important to visualize them. The code for visualizing DERs can easily be adapted to visualize other regions. Although, the width and number of the regions will influence the computing resources needed to make the plots.</p>
<p>Because the unscaled BigWig files are available in recount2, several visualization packages can be used such as <code>epivizr</code> <span class="citation">(26)</span>, <code>wiggleplotr</code> <span class="citation">(27)</span> and <code>derfinderPlot</code> <span class="citation">(9)</span>. With all of them it is important to remember to scale the data except when visualizing the mean BigWig file for a given project.</p>
<p>First, we need to get the list of URLs for the BigWig files. We can either manually construct them or search them inside the <code>recount_url</code> table. For this workshop, we have the data locally so we will use those files.</p>
<div class="sourceCode" id="cb217"><pre class="downlit">
<span class="co">## Construct the list of BigWig URLs</span>
<span class="co">## They have the following form:</span>
<span class="co">## http://duffel.rail.bio/recount/</span>
<span class="co">## project id</span>
<span class="co">## /bw/</span>
<span class="co">## sample run id</span>
<span class="co">## .bw</span>
<span class="kw">bws_web</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(
    <span class="st">"http://duffel.rail.bio/recount/SRP056604/bw/"</span>,
    <span class="fu">colData</span>(<span class="kw">rse_er</span>)<span class="op">$</span><span class="kw">bigwig_file</span>
)

<span class="co">## Note that they are also present in the recount_url data.frame</span>
<span class="kw">bws_url</span> <span class="op">&lt;-</span> <span class="kw">recount_url</span><span class="op">$</span><span class="kw">url</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(
    <span class="fu">colData</span>(<span class="kw">rse_er</span>)<span class="op">$</span><span class="kw">bigwig_file</span>,
    <span class="kw">recount_url</span><span class="op">$</span><span class="kw">file_name</span>
)]
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="kw">bws_web</span>, <span class="kw">bws_url</span>)
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb219"><pre class="downlit">
<span class="co">## Local bigwigs</span>
<span class="kw">bws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">local_path</span>, <span class="st">"bw"</span>, <span class="fu">colData</span>(<span class="kw">rse_er</span>)<span class="op">$</span><span class="kw">bigwig_file</span>)
<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="kw">bws</span>))
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb221"><pre class="downlit">
<span class="co">## Use the sample run ids as the sample names</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">bws</span>) <span class="op">&lt;-</span> <span class="fu">colData</span>(<span class="kw">rse_er</span>)<span class="op">$</span><span class="kw">run</span>
</pre></div>
<p>We will visualize the DERs using <code>derfinderPlot</code>, similar to what was done in Jaffe et al., 2015 <span class="citation">(28)</span>. We will first add a little padding to the regions: 100 base-pairs on each side.</p>
<div class="sourceCode" id="cb222"><pre class="downlit">
<span class="co">## Add 100 bp padding on each side</span>
<span class="kw">regions_resized</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">resize</a></span>(<span class="kw">regions_by_padj</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>],
    <span class="fu">width</span>(<span class="kw">regions_by_padj</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>]) <span class="op">+</span> <span class="fl">200</span>,
    fix = <span class="st">"center"</span>
)
</pre></div>
<p>Next, we obtain the base-pair coverage data for each DER and scale the data to a library size of 40 million 100 base-pair reads using the coverage AUC information we have in the metadata.</p>
<div class="sourceCode" id="cb223"><pre class="downlit">
<span class="co">## Get the bp coverage data for the plots</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/lcolladotor/derfinder">"derfinder"</a></span>)
<span class="kw">regionCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html">getRegionCoverage</a></span>(
    regions = <span class="kw">regions_resized</span>, files = <span class="kw">bws</span>,
    targetSize = <span class="fl">40</span> <span class="op">*</span> <span class="fl">1e6</span> <span class="op">*</span> <span class="fl">100</span>, totalMapped = <span class="fu">colData</span>(<span class="kw">rse_er</span>)<span class="op">$</span><span class="kw">auc</span>,
    verbose = <span class="fl">FALSE</span>
)
</pre></div>
<p>The function <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> requires several pieces of annotation information for the plots that use a TxDb object. For recount2 we used Gencode v25 hg38’s annotation, which means that we need to process it manually instead of using a pre-computed TxDb package.</p>
<p>To create a TxDb object for Gencode v25, this can take several lines of code and to simplify this process we created the <code>GenomicState</code> package which provides processed annotation files for Gencode that can be easily used with <code>derfinder</code> and other Bioconductor packages. <code>GenomicState</code> is powered by <code>AnnotationHub</code> and thus uses similar syntax to it. Although years later, we created <code>GenomicState</code> in response to a comment from one of the reviewers of the <code>recountWorkflow</code> paper.</p>
<div class="sourceCode" id="cb224"><pre class="downlit">
<span class="co">## Get the chromosome information for hg38</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/LieberInstitute/GenomicState">"GenomicState"</a></span>)
<span class="kw">gencode_v25_hg38_txdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html">GenomicStateHub</a></span>(
    version = <span class="st">"25"</span>,
    genome = <span class="st">"hg38"</span>,
    filetype = <span class="st">"TxDb"</span>
)[[<span class="fl">1</span>]]
</pre></div>
<pre><code>## using temporary cache /tmp/RtmpKDI1FD/BiocFileCache</code></pre>
<pre><code>## snapshotDate(): 2020-07-20</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb230"><pre class="downlit">
<span class="co">## Explore the TxDb object</span>
<span class="kw">gencode_v25_hg38_txdb</span>
</pre></div>
<pre><code>## TxDb object:
## # Db type: TxDb
## # Supporting package: GenomicFeatures
## # Data source: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_25/gencode.v25.annotation.gtf.gz
## # Organism: Homo sapiens
## # Taxonomy ID: 9606
## # miRBase build ID: NA
## # Genome: hg38
## # transcript_nrow: 198093
## # exon_nrow: 678580
## # cds_nrow: 270623
## # Db created by: GenomicFeatures package from Bioconductor
## # Creation time: 2019-10-07 09:59:57 -0400 (Mon, 07 Oct 2019)
## # GenomicFeatures version at creation time: 1.36.4
## # RSQLite version at creation time: 2.1.2
## # DBSCHEMAVERSION: 1.2</code></pre>
<p>Now that we have a TxDb object for Gencode v25 on hg38 coordinates, we can use <code>bumphunter</code>’s <span class="citation">(29)</span> annotation functions for annotating the original 10 regions we were working with. Since we are using Ensembl instead of Entrez gene ids, we need to pass this information to <code><a href="https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html">annotateTranscripts()</a></code>. Otherwise, the function will fail to retrieve the gene symbols.</p>
<div class="sourceCode" id="cb232"><pre class="downlit">
<span class="co">## Download annotated transcripts for gencode v25 based on the TxDb object</span>
<span class="co">## we built previously.</span>
<span class="kw">ann_gencode_v25_hg38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html">GenomicStateHub</a></span>(
    version = <span class="st">"25"</span>,
    genome = <span class="st">"hg38"</span>,
    filetype = <span class="st">"AnnotatedGenes"</span>
)[[<span class="fl">1</span>]]
</pre></div>
<pre><code>## snapshotDate(): 2020-07-20</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb237"><pre class="downlit">
<span class="co">## Annotate the regions of interest</span>
<span class="co">## Note that we are using the original regions, not the resized ones</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/ririzarr/bumphunter">"bumphunter"</a></span>)
<span class="kw">nearest_ann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html">matchGenes</a></span>(<span class="kw">regions_by_padj</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], <span class="kw">ann_gencode_v25_hg38</span>)
</pre></div>
<p>The final piece we need to run <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> is information about which base-pairs are exonic, intronic, etc. This is done via the <code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions()</a></code> function in <code>derfinder</code>, which itself requires prior processing of the TxDb information by <code><a href="https://rdrr.io/pkg/derfinder/man/makeGenomicState.html">makeGenomicState()</a></code>.</p>
<div class="sourceCode" id="cb238"><pre class="downlit">
<span class="co">## Download the genomic state object using the gencode TxDb object</span>
<span class="kw">gs_gencode_v25_hg38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html">GenomicStateHub</a></span>(
    version = <span class="st">"25"</span>,
    genome = <span class="st">"hg38"</span>,
    filetype = <span class="st">"GenomicState"</span>
)[[<span class="fl">1</span>]]
</pre></div>
<pre><code>## snapshotDate(): 2020-07-20</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb243"><pre class="downlit">
<span class="co">## Annotate the original regions</span>
<span class="kw">regions_ann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions</a></span>(
    <span class="kw">regions_resized</span>,
    <span class="kw">gs_gencode_v25_hg38</span><span class="op">$</span><span class="kw">fullGenome</span>
)
</pre></div>
<pre><code>## 2020-07-26 21:30:30 annotateRegions: counting</code></pre>
<pre><code>## 2020-07-26 21:30:31 annotateRegions: annotating</code></pre>
<p>We can finally use <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> to visualize the top 10 regions coloring by whether they are LOAD or control samples. Known exons are shown in dark blue, introns in light blue.</p>
<div class="sourceCode" id="cb246"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/leekgroup/derfinderPlot">"derfinderPlot"</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage</a></span>(
    regions = <span class="kw">regions_resized</span>, regionCoverage = <span class="kw">regionCov</span>,
    groupInfo = <span class="fu">colData</span>(<span class="kw">rse_er</span>)<span class="op">$</span><span class="kw">case</span>,
    nearestAnnotation = <span class="kw">nearest_ann</span>,
    annotatedRegions = <span class="kw">regions_ann</span>,
    txdb = <span class="kw">gencode_v25_hg38_txdb</span>,
    scalefac = <span class="fl">1</span>, ylab = <span class="st">"Coverage (RP40M, 100bp)"</span>,
    ask = <span class="fl">FALSE</span>, verbose = <span class="fl">FALSE</span>
)
</pre></div>
<p><img src="recount-workshop_files/figure-html/region_plots-1.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-2.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-3.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-4.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-5.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-6.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-7.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-8.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-9.png" width="100%" style="display: block; margin: auto;"><img src="recount-workshop_files/figure-html/region_plots-10.png" width="100%" style="display: block; margin: auto;"></p>
<p>In the previous plots we can see that some DERs are longer than known exons, others match known exons, and some are un-annotated expressed regions.</p>
</div>
</div>
<div id="summary" class="section level2">
<h2 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h2>
<p>In this workshop we described in detail the available data in recount2, how the coverage count matrices were computed, the metadata included in recount2 and how to get new phenotypic information from other sources including from <code>recount-brain</code>. We showed how to perform a differential expression analysis at the gene and exon levels as well as use an annotation-agnostic approach. Finally, we explained how to visualize the base-pair information for a given set of regions using <code>derfinder</code> and <code>GenomicState</code>. This workshop constitutes a strong basis to leverage the recount2 and <code>recount-brain</code> data for human RNA-seq analyses.</p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb247"><pre class="downlit">
<span class="co">## Pandoc information</span>
<span class="kw">rmarkdown</span>::<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/pandoc_available.html">pandoc_version</a></span>()
</pre></div>
<pre><code>## [1] '2.7.3'</code></pre>
<div class="sourceCode" id="cb249"><pre class="downlit">
<span class="co">## Time for reproducing this workflow, in minutes</span>
<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>()[<span class="fl">3</span>] <span class="op">/</span> <span class="fl">60</span>, <span class="fl">1</span>)
</pre></div>
<pre><code>## elapsed 
##     6.9</code></pre>
<div class="sourceCode" id="cb251"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(width = <span class="fl">100</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/r-lib/sessioninfo#readme">"sessioninfo"</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span>()
</pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.0 (2020-04-24)
##  os       Ubuntu 18.04.4 LTS          
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       UTC                         
##  date     2020-07-26                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────
##  package                * version    date       lib source                        
##  acepack                  1.4.1      2016-10-29 [1] RSPM (R 4.0.0)                
##  annotate                 1.67.0     2020-04-27 [1] Bioconductor                  
##  AnnotationDbi          * 1.51.3     2020-07-25 [1] Bioconductor                  
##  AnnotationFilter         1.13.0     2020-04-27 [1] Bioconductor                  
##  AnnotationForge          1.31.2     2020-06-01 [1] Bioconductor                  
##  AnnotationHub          * 2.21.1     2020-06-15 [1] Bioconductor                  
##  askpass                  1.1        2019-01-13 [2] RSPM (R 4.0.0)                
##  assertthat               0.2.1      2019-03-21 [2] RSPM (R 4.0.0)                
##  backports                1.1.7      2020-05-13 [2] RSPM (R 4.0.0)                
##  base64enc                0.1-3      2015-07-28 [1] RSPM (R 4.0.0)                
##  BiasedUrn                1.07       2015-12-28 [1] RSPM (R 4.0.0)                
##  bibtex                   0.4.2.2    2020-01-02 [1] RSPM (R 4.0.0)                
##  Biobase                * 2.49.0     2020-04-27 [1] Bioconductor                  
##  BiocFileCache          * 1.13.0     2020-04-27 [1] Bioconductor                  
##  BiocGenerics           * 0.35.4     2020-06-04 [1] Bioconductor                  
##  BiocManager              1.30.10    2019-11-16 [2] CRAN (R 4.0.0)                
##  BiocParallel             1.23.2     2020-07-06 [1] Bioconductor                  
##  BiocStyle                2.17.0     2020-04-27 [1] Bioconductor                  
##  BiocVersion              3.12.0     2020-04-27 [2] Bioconductor                  
##  biomaRt                  2.45.2     2020-07-05 [1] Bioconductor                  
##  Biostrings               2.57.2     2020-06-09 [1] Bioconductor                  
##  biovizBase               1.37.0     2020-04-27 [1] Bioconductor                  
##  bit                      1.1-15.2   2020-02-10 [1] RSPM (R 4.0.0)                
##  bit64                    0.9-7      2017-05-08 [1] RSPM (R 4.0.0)                
##  bitops                   1.0-6      2013-08-17 [1] RSPM (R 4.0.0)                
##  blob                     1.2.1      2020-01-20 [1] RSPM (R 4.0.0)                
##  bookdown                 0.19       2020-05-15 [1] RSPM (R 4.0.0)                
##  BSgenome                 1.57.5     2020-07-22 [1] Bioconductor                  
##  bumphunter             * 1.31.0     2020-04-28 [1] Bioconductor                  
##  Category                 2.55.0     2020-04-27 [1] Bioconductor                  
##  caTools                  1.18.0     2020-01-17 [1] RSPM (R 4.0.0)                
##  checkmate                2.0.0      2020-02-06 [1] RSPM (R 4.0.0)                
##  cli                      2.0.2      2020-02-28 [2] RSPM (R 4.0.0)                
##  cluster                  2.1.0      2019-06-19 [3] CRAN (R 4.0.0)                
##  clusterProfiler        * 3.17.0     2020-04-27 [1] Bioconductor                  
##  codetools                0.2-16     2018-12-24 [3] CRAN (R 4.0.0)                
##  colorspace               1.4-1      2019-03-18 [1] RSPM (R 4.0.0)                
##  cowplot                  1.0.0      2019-07-11 [1] RSPM (R 4.0.0)                
##  crayon                   1.3.4      2017-09-16 [2] RSPM (R 4.0.0)                
##  crosstalk                1.1.0.1    2020-03-13 [2] RSPM (R 4.0.0)                
##  curl                     4.3        2019-12-02 [2] RSPM (R 4.0.0)                
##  data.table               1.12.8     2019-12-09 [1] RSPM (R 4.0.0)                
##  DBI                      1.1.0      2019-12-15 [1] RSPM (R 4.0.0)                
##  dbplyr                 * 1.4.4      2020-05-27 [1] RSPM (R 4.0.0)                
##  DEFormats                1.17.0     2020-04-27 [1] Bioconductor                  
##  DelayedArray           * 0.15.7     2020-07-14 [1] Bioconductor                  
##  dendextend               1.13.4     2020-02-28 [1] RSPM (R 4.0.0)                
##  derfinder              * 1.23.1     2020-05-10 [1] Bioconductor                  
##  derfinderHelper          1.23.1     2020-05-10 [1] Bioconductor                  
##  derfinderPlot          * 1.23.1     2020-05-10 [1] Bioconductor                  
##  desc                     1.2.0      2018-05-01 [2] RSPM (R 4.0.0)                
##  DESeq2                 * 1.29.7     2020-07-16 [1] Bioconductor                  
##  dichromat                2.0-0      2013-01-24 [1] RSPM (R 4.0.0)                
##  digest                   0.6.25     2020-02-23 [2] RSPM (R 4.0.0)                
##  DO.db                    2.9        2020-07-25 [1] Bioconductor                  
##  doParallel               1.0.15     2019-08-02 [1] RSPM (R 4.0.0)                
##  doRNG                    1.8.2      2020-01-27 [1] RSPM (R 4.0.0)                
##  DOSE                     3.15.0     2020-04-27 [1] Bioconductor                  
##  downloader               0.4        2015-07-09 [1] RSPM (R 4.0.0)                
##  dplyr                    1.0.0      2020-05-29 [1] RSPM (R 4.0.0)                
##  DT                       0.13       2020-03-23 [2] RSPM (R 4.0.0)                
##  edgeR                    3.31.4     2020-06-10 [1] Bioconductor                  
##  ellipsis                 0.3.1      2020-05-15 [2] RSPM (R 4.0.0)                
##  enrichplot               1.9.1      2020-04-29 [1] Bioconductor                  
##  ensembldb                2.13.1     2020-05-06 [1] Bioconductor                  
##  europepmc                0.4        2020-05-31 [1] RSPM (R 4.0.0)                
##  evaluate                 0.14       2019-05-28 [2] RSPM (R 4.0.0)                
##  fansi                    0.4.1      2020-01-08 [2] RSPM (R 4.0.0)                
##  farver                   2.0.3      2020-01-16 [1] RSPM (R 4.0.0)                
##  fastmap                  1.0.1      2019-10-08 [1] RSPM (R 4.0.0)                
##  fastmatch                1.1-0      2017-01-28 [1] RSPM (R 4.0.0)                
##  fdrtool                  1.2.15     2015-07-08 [1] RSPM (R 4.0.0)                
##  fgsea                    1.15.0     2020-04-27 [1] Bioconductor                  
##  foreach                * 1.5.0      2020-03-30 [1] RSPM (R 4.0.0)                
##  foreign                  0.8-78     2020-04-13 [3] CRAN (R 4.0.0)                
##  Formula                  1.2-3      2018-05-03 [1] RSPM (R 4.0.0)                
##  fs                       1.4.1      2020-04-04 [2] RSPM (R 4.0.0)                
##  gclus                    1.3.2      2019-01-07 [1] RSPM (R 4.0.0)                
##  gdata                    2.18.0     2017-06-06 [1] RSPM (R 4.0.0)                
##  genefilter               1.71.0     2020-04-27 [1] Bioconductor                  
##  geneLenDataBase          1.25.0     2020-04-30 [1] Bioconductor                  
##  geneplotter              1.67.0     2020-04-27 [1] Bioconductor                  
##  generics                 0.0.2      2018-11-29 [1] RSPM (R 4.0.0)                
##  GenomeInfoDb           * 1.25.8     2020-07-03 [1] Bioconductor                  
##  GenomeInfoDbData         1.2.3      2020-07-25 [1] Bioconductor                  
##  GenomicAlignments        1.25.3     2020-06-09 [1] Bioconductor                  
##  GenomicFeatures        * 1.41.2     2020-07-07 [1] Bioconductor                  
##  GenomicFiles             1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomicRanges          * 1.41.5     2020-06-09 [1] Bioconductor                  
##  GenomicState           * 0.99.9     2020-07-26 [1] Bioconductor                  
##  GEOquery                 2.57.0     2020-04-27 [1] Bioconductor                  
##  GGally                   2.0.0      2020-06-06 [1] RSPM (R 4.0.0)                
##  ggbio                    1.37.0     2020-04-27 [1] Bioconductor                  
##  ggforce                  0.3.1      2019-08-20 [1] RSPM (R 4.0.0)                
##  ggplot2                  3.3.1      2020-05-28 [1] RSPM (R 4.0.0)                
##  ggplotify                0.0.5      2020-03-12 [1] RSPM (R 4.0.0)                
##  ggraph                   2.0.3      2020-05-20 [1] RSPM (R 4.0.0)                
##  ggrepel                  0.8.2      2020-03-08 [1] RSPM (R 4.0.0)                
##  ggridges                 0.5.2      2020-01-12 [1] RSPM (R 4.0.0)                
##  glue                     1.4.1      2020-05-13 [2] RSPM (R 4.0.0)                
##  GO.db                  * 3.11.4     2020-07-25 [1] Bioconductor                  
##  GOSemSim                 2.15.1     2020-07-25 [1] Bioconductor                  
##  goseq                    1.41.0     2020-04-27 [1] Bioconductor                  
##  GOstats                  2.55.0     2020-04-27 [1] Bioconductor                  
##  gplots                 * 3.0.3      2020-02-25 [1] RSPM (R 4.0.0)                
##  graph                  * 1.67.1     2020-05-27 [1] Bioconductor                  
##  graphlayouts             0.7.0      2020-04-25 [1] RSPM (R 4.0.0)                
##  gridBase                 0.4-7      2014-02-24 [1] RSPM (R 4.0.0)                
##  gridExtra                2.3        2017-09-09 [1] RSPM (R 4.0.0)                
##  gridGraphics             0.5-0      2020-02-25 [1] RSPM (R 4.0.0)                
##  GSEABase                 1.51.1     2020-05-29 [1] Bioconductor                  
##  gtable                   0.3.0      2019-03-25 [1] RSPM (R 4.0.0)                
##  gtools                   3.8.2      2020-03-31 [1] RSPM (R 4.0.0)                
##  heatmaply                1.1.0      2020-03-28 [1] RSPM (R 4.0.0)                
##  highr                    0.8        2019-03-20 [2] RSPM (R 4.0.0)                
##  Hmisc                    4.4-0      2020-03-23 [1] RSPM (R 4.0.0)                
##  hms                      0.5.3      2020-01-08 [1] RSPM (R 4.0.0)                
##  htmlTable                1.13.3     2019-12-04 [1] RSPM (R 4.0.0)                
##  htmltools                0.4.0      2019-10-04 [2] RSPM (R 4.0.0)                
##  htmlwidgets              1.5.1      2019-10-08 [2] RSPM (R 4.0.0)                
##  httpuv                   1.5.4      2020-06-06 [1] RSPM (R 4.0.0)                
##  httr                     1.4.1      2019-08-05 [2] RSPM (R 4.0.0)                
##  ideal                  * 1.13.1     2020-05-26 [1] Bioconductor                  
##  igraph                   1.2.5      2020-03-19 [1] CRAN (R 4.0.0)                
##  IHW                      1.17.0     2020-04-27 [1] Bioconductor                  
##  interactiveDisplayBase   1.27.5     2020-06-01 [1] Bioconductor                  
##  IRanges                * 2.23.10    2020-06-13 [1] Bioconductor                  
##  iterators              * 1.0.12     2019-07-26 [1] RSPM (R 4.0.0)                
##  jpeg                     0.1-8.1    2019-10-24 [1] RSPM (R 4.0.0)                
##  jsonlite                 1.6.1      2020-02-02 [2] RSPM (R 4.0.0)                
##  KernSmooth               2.23-16    2019-10-15 [3] CRAN (R 4.0.0)                
##  knitcitations            1.0.10     2019-09-15 [1] RSPM (R 4.0.0)                
##  knitr                    1.28       2020-02-06 [2] RSPM (R 4.0.0)                
##  knitrBootstrap           1.0.2      2018-05-24 [1] RSPM (R 4.0.0)                
##  labeling                 0.3        2014-08-23 [1] RSPM (R 4.0.0)                
##  later                    1.1.0.1    2020-06-05 [2] RSPM (R 4.0.0)                
##  lattice                  0.20-41    2020-04-02 [3] CRAN (R 4.0.0)                
##  latticeExtra             0.6-29     2019-12-19 [1] RSPM (R 4.0.0)                
##  lazyeval                 0.2.2      2019-03-15 [2] RSPM (R 4.0.0)                
##  lifecycle                0.2.0      2020-03-06 [2] RSPM (R 4.0.0)                
##  limma                    3.45.9     2020-07-14 [1] Bioconductor                  
##  locfit                 * 1.5-9.4    2020-03-25 [1] RSPM (R 4.0.0)                
##  lpsymphony               1.17.0     2020-04-27 [1] Bioconductor (R 4.0.0)        
##  lubridate                1.7.8      2020-04-06 [1] RSPM (R 4.0.0)                
##  magrittr                 1.5        2014-11-22 [2] RSPM (R 4.0.0)                
##  markdown                 1.1        2019-08-07 [2] RSPM (R 4.0.0)                
##  MASS                     7.3-51.5   2019-12-20 [3] CRAN (R 4.0.0)                
##  Matrix                 * 1.2-18     2019-11-27 [3] CRAN (R 4.0.0)                
##  matrixStats            * 0.56.0     2020-03-13 [1] RSPM (R 4.0.0)                
##  memoise                  1.1.0      2017-04-21 [2] RSPM (R 4.0.0)                
##  mgcv                     1.8-31     2019-11-09 [3] CRAN (R 4.0.0)                
##  mime                     0.9        2020-02-04 [2] RSPM (R 4.0.0)                
##  munsell                  0.5.0      2018-06-12 [1] RSPM (R 4.0.0)                
##  nlme                     3.1-147    2020-04-13 [3] CRAN (R 4.0.0)                
##  NMF                      0.22.0     2020-02-12 [1] RSPM (R 4.0.0)                
##  nnet                     7.3-13     2020-02-25 [3] CRAN (R 4.0.0)                
##  openssl                  1.4.1      2019-07-18 [2] RSPM (R 4.0.0)                
##  org.Hs.eg.db           * 3.11.4     2020-07-25 [1] Bioconductor                  
##  OrganismDbi              1.31.0     2020-04-27 [1] Bioconductor                  
##  pcaExplorer              2.15.2     2020-06-06 [1] Bioconductor                  
##  pheatmap                 1.0.12     2019-01-04 [1] RSPM (R 4.0.0)                
##  pillar                   1.4.4      2020-05-05 [2] RSPM (R 4.0.0)                
##  pkgconfig                2.0.3      2019-09-22 [2] RSPM (R 4.0.0)                
##  pkgdown                  1.5.1.9000 2020-07-25 [1] Github (r-lib/pkgdown@eae56f0)
##  pkgmaker                 0.31.1     2020-03-19 [1] RSPM (R 4.0.0)                
##  plotly                   4.9.2.1    2020-04-04 [1] RSPM (R 4.0.0)                
##  plyr                     1.8.6      2020-03-03 [1] RSPM (R 4.0.0)                
##  png                      0.1-7      2013-12-03 [1] RSPM (R 4.0.0)                
##  polyclip                 1.10-0     2019-03-14 [1] RSPM (R 4.0.0)                
##  prettyunits              1.1.1      2020-01-24 [2] RSPM (R 4.0.0)                
##  progress                 1.2.2      2019-05-16 [1] RSPM (R 4.0.0)                
##  promises                 1.1.0      2019-10-04 [2] RSPM (R 4.0.0)                
##  ProtGenerics             1.21.0     2020-04-27 [1] Bioconductor                  
##  purrr                    0.3.4      2020-04-17 [2] RSPM (R 4.0.0)                
##  qvalue                   2.21.0     2020-04-27 [1] Bioconductor                  
##  R6                       2.4.1      2019-11-12 [2] RSPM (R 4.0.0)                
##  rappdirs                 0.3.1      2016-03-28 [1] RSPM (R 4.0.0)                
##  RBGL                     1.65.0     2020-04-27 [1] Bioconductor                  
##  RColorBrewer             1.1-2      2014-12-07 [1] RSPM (R 4.0.0)                
##  Rcpp                     1.0.4.6    2020-04-09 [2] RSPM (R 4.0.0)                
##  RCurl                    1.98-1.2   2020-04-18 [1] RSPM (R 4.0.0)                
##  readr                    1.3.1      2018-12-21 [1] RSPM (R 4.0.0)                
##  recount                * 1.15.1     2020-05-12 [1] Bioconductor                  
##  recountWorkshop2020    * 0.99.0     2020-07-26 [1] Bioconductor                  
##  RefManageR               1.2.12     2019-04-03 [1] RSPM (R 4.0.0)                
##  regionReport           * 1.23.2     2020-05-10 [1] Bioconductor                  
##  registry                 0.5-1      2019-03-05 [1] RSPM (R 4.0.0)                
##  rentrez                  1.2.2      2019-05-02 [1] RSPM (R 4.0.0)                
##  reshape                  0.8.8      2018-10-23 [1] RSPM (R 4.0.0)                
##  reshape2                 1.4.4      2020-04-09 [1] RSPM (R 4.0.0)                
##  Rgraphviz                2.33.0     2020-04-27 [1] Bioconductor                  
##  rintrojs                 0.2.2      2019-05-29 [1] RSPM (R 4.0.0)                
##  rlang                    0.4.6      2020-05-02 [2] RSPM (R 4.0.0)                
##  rmarkdown                2.2        2020-05-31 [1] RSPM (R 4.0.0)                
##  rngtools                 1.5        2020-01-23 [1] RSPM (R 4.0.0)                
##  rpart                    4.1-15     2019-04-12 [3] CRAN (R 4.0.0)                
##  rprojroot                1.3-2      2018-01-03 [2] RSPM (R 4.0.0)                
##  Rsamtools                2.5.3      2020-06-26 [1] Bioconductor                  
##  RSQLite                  2.2.0      2020-01-07 [1] RSPM (R 4.0.0)                
##  rstudioapi               0.11       2020-02-07 [2] RSPM (R 4.0.0)                
##  rtracklayer            * 1.49.4     2020-07-24 [1] Bioconductor                  
##  rvcheck                  0.1.8      2020-03-01 [1] RSPM (R 4.0.0)                
##  S4Vectors              * 0.27.12    2020-06-09 [1] Bioconductor                  
##  scales                   1.1.1      2020-05-11 [1] RSPM (R 4.0.0)                
##  scatterpie               0.1.4      2019-11-08 [1] RSPM (R 4.0.0)                
##  seriation                1.2-8      2019-08-27 [1] RSPM (R 4.0.0)                
##  sessioninfo            * 1.1.1      2018-11-05 [2] RSPM (R 4.0.0)                
##  shiny                    1.4.0.2    2020-03-13 [1] RSPM (R 4.0.0)                
##  shinyAce                 0.4.1      2019-09-24 [1] RSPM (R 4.0.0)                
##  shinyBS                  0.61       2015-03-31 [1] RSPM (R 4.0.0)                
##  shinydashboard           0.7.1      2018-10-17 [1] RSPM (R 4.0.0)                
##  slam                     0.1-47     2019-12-21 [1] RSPM (R 4.0.0)                
##  SparseM                * 1.78       2019-12-13 [1] RSPM (R 4.0.0)                
##  stringi                  1.4.6      2020-02-17 [2] RSPM (R 4.0.0)                
##  stringr                  1.4.0      2019-02-10 [2] RSPM (R 4.0.0)                
##  SummarizedExperiment   * 1.19.6     2020-07-09 [1] Bioconductor                  
##  survival                 3.1-12     2020-04-10 [3] CRAN (R 4.0.0)                
##  threejs                  0.3.3      2020-01-21 [1] RSPM (R 4.0.0)                
##  tibble                   3.0.1      2020-04-20 [2] RSPM (R 4.0.0)                
##  tidygraph                1.2.0      2020-05-12 [1] RSPM (R 4.0.0)                
##  tidyr                    1.1.0      2020-05-20 [1] RSPM (R 4.0.0)                
##  tidyselect               1.1.0      2020-05-11 [1] RSPM (R 4.0.0)                
##  topGO                  * 2.41.0     2020-04-27 [1] Bioconductor                  
##  triebeard                0.3.0      2016-08-04 [1] RSPM (R 4.0.0)                
##  TSP                      1.1-10     2020-04-17 [1] RSPM (R 4.0.0)                
##  tweenr                   1.0.1      2018-12-14 [1] RSPM (R 4.0.0)                
##  UpSetR                   1.4.0      2019-05-22 [1] RSPM (R 4.0.0)                
##  urltools                 1.7.3      2019-04-14 [1] RSPM (R 4.0.0)                
##  VariantAnnotation        1.35.3     2020-06-26 [1] Bioconductor                  
##  vctrs                    0.3.1      2020-06-05 [2] RSPM (R 4.0.0)                
##  viridis                  0.5.1      2018-03-29 [1] RSPM (R 4.0.0)                
##  viridisLite              0.3.0      2018-02-01 [1] RSPM (R 4.0.0)                
##  webshot                  0.5.2      2019-11-22 [1] RSPM (R 4.0.0)                
##  withr                    2.2.0      2020-04-20 [2] RSPM (R 4.0.0)                
##  xfun                     0.14       2020-05-20 [2] RSPM (R 4.0.0)                
##  XML                      3.99-0.3   2020-01-20 [1] RSPM (R 4.0.0)                
##  xml2                     1.3.2      2020-04-23 [2] RSPM (R 4.0.0)                
##  xtable                   1.8-4      2019-04-21 [1] RSPM (R 4.0.0)                
##  XVector                  0.29.3     2020-06-25 [1] Bioconductor                  
##  yaml                     2.2.1      2020-02-01 [2] RSPM (R 4.0.0)                
##  zlibbioc                 1.35.0     2020-04-27 [1] Bioconductor                  
## 
## [1] /usr/local/lib/R/host-site-library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="author-contributions" class="section level2">
<h2 class="hasAnchor">
<a href="#author-contributions" class="anchor"></a>Author contributions</h2>
<p>LCT, AN and AEJ wrote the workflow from which this workshop is adapted. LCT wrote this workshop.</p>
</div>
<div id="competing-interests" class="section level2">
<h2 class="hasAnchor">
<a href="#competing-interests" class="anchor"></a>Competing interests</h2>
<p>No competing interests were disclosed.</p>
</div>
<div id="grant-information" class="section level2">
<h2 class="hasAnchor">
<a href="#grant-information" class="anchor"></a>Grant information</h2>
<p>LCT and AEJ were supported by NIH grant R21 MH109956-01.</p>
</div>
<div id="acknowledgments" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h2>
<p>We would like to acknowledge the Andrew Jaffe and Alexis Battle lab members for feedback on the explanatory pictures and the written recountWorkflow manuscript.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-recountWorkflow">
<p>1. Collado-Torres L, Nellore A, Jaffe AE. Recount workflow: Accessing over 70,000 human rna-seq samples with bioconductor [version 1; referees: 1 approved, 2 approved with reservations]. F1000Research [Internet]. 2017; Available from: <a href="https://f1000research.com/articles/6-1558/v1">https://f1000research.com/articles/6-1558/v1</a></p>
</div>
<div id="ref-huber2015">
<p>2. Huber W, Carey VJ, Gentleman R, Anders S, Carlson M, Carvalho BS, et al. Orchestrating high-throughput genomic analysis with bioconductor. Nature methods. 2015;12(2):115–21. </p>
</div>
<div id="ref-Law2016a">
<p>3. Law CW, Alhamdoosh M, Su S, Smyth GK, Ritchie ME. RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR. F1000Research. 2016;5(May):1408. </p>
</div>
<div id="ref-Love2016">
<p>4. Love MI, Anders S, Kim V, Huber W. RNA-Seq workflow: gene-level exploratory analysis and differential expression. F1000Research. 2016;4(May):1070. </p>
</div>
<div id="ref-Chen2016">
<p>5. Chen Y, Lun ATL, Smyth GK. From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline. F1000Research. 2016;5(May):1438. </p>
</div>
<div id="ref-frazee2011">
<p>6. Frazee AC, Langmead B, Leek JT. ReCount: A multi-experiment resource of analysis-ready RNA-seq gene count datasets. BMC bioinformatics. 2011;12:449. </p>
</div>
<div id="ref-airway">
<p>7. Himes, E. B, Jiang, X., Wagner, P., et al. RNA-Seq Transcriptome Profiling Identifies CRISPLD2 as a Glucocorticoid Responsive Gene that Modulates Cytokine Function in Airway Smooth Muscle Cells. PLoS ONE. 2014;9(6):e99625. </p>
</div>
<div id="ref-collado-torres_reproducible_2017">
<p>8. Collado-Torres L, Nellore A, Kammers K, Ellis SE, Taub MA, Hansen KD, et al. Reproducible RNA-seq analysis using recount2. Nature Biotechnology. 2017 Apr;35(4):319–21. </p>
</div>
<div id="ref-collado-torres_flexible_2017">
<p>9. Collado-Torres L, Nellore A, Frazee AC, Wilks C, Love MI, Langmead B, et al. Flexible expressed region analysis for RNA-seq with derfinder. Nucleic Acids Research. 2017 Jan;45(2):e9–9. </p>
</div>
<div id="ref-summarizedexperiment">
<p>10. Morgan M, Obenchain V, Hester J, Pagès H. SummarizedExperiment: SummarizedExperiment container. 2019. </p>
</div>
<div id="ref-Wilks2018">
<p>11. Wilks C, Gaddipati P, Nellore A, Langmead B. Snaptron: Querying splicing patterns across tens of thousands of RNA-seq samples. Bioinformatics [Internet]. 2018 Jan [cited 2019 May 29];34(1):114–6. Available from: <a href="https://academic.oup.com/bioinformatics/article/34/1/114/4101942">https://academic.oup.com/bioinformatics/article/34/1/114/4101942</a></p>
</div>
<div id="ref-rail-rna">
<p>12. Nellore A, Collado-Torres L, Jaffe AE, Alquicira-Hernández J, Wilks C, Pritt J, et al. Rail-RNA: Scalable analysis of RNA-seq splicing and coverage. Bioinformatics (Oxford, England). 2016 Sep; </p>
</div>
<div id="ref-lawrence2013">
<p>13. Lawrence M, Huber W, Pages H, Aboyoun P, Carlson M, Gentleman R, et al. Software for computing and annotating genomic ranges. PLoS computational biology. 2013;9(8):e1003118. </p>
</div>
<div id="ref-magistri2015">
<p>14. Magistri M, Velmeshev D, Makhmutova M, Faghihi MA. Transcriptomics Profiling of Alzheimer’s Disease Reveal Neurovascular Defects, Altered Amyloid-β Homeostasis, and Deregulated Expression of Long Noncoding RNAs. Journal of Alzheimer’s disease: JAD. 2015;48(3):647–65. </p>
</div>
<div id="ref-Ellis2018">
<p>15. Ellis SE, Collado-Torres L, Jaffe AE, Leek JT. Improving the value of public rna-seq expression data by phenotype prediction. Nucl Acids Res [Internet]. 2018; Available from: <a href="https://doi.org/10.1093/nar/gky102">https://doi.org/10.1093/nar/gky102</a></p>
</div>
<div id="ref-Razmara2019">
<p>16. Razmara A, Ellis SE, Sokolowski DJ, Davis S, Wilson MD, Leek JT, et al. Recount-brain: A curated repository of human brain rna-seq datasets metadata. bioRxiv [Internet]. 2019; Available from: <a href="https://doi.org/10.1101/618025">https://doi.org/10.1101/618025</a></p>
</div>
<div id="ref-TCGAbiolinks">
<p>17. Colaprico A, Silva TC, Olsen C, Garofano L, Cava C, Garolini D, et al. TCGAbiolinks: An r/bioconductor package for integrative analysis of tcga data. Nucleic Acids Research. 2015; </p>
</div>
<div id="ref-love2014moderated">
<p>18. Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for rna-seq data with deseq2. Genome biology. 2014;15(12):1–21. </p>
</div>
<div id="ref-ideal">
<p>19. Marini F. Ideal: Interactive differential expression analysis [Internet]. 2018. Available from: <a href="http://bioconductor.org/packages/ideal/">http://bioconductor.org/packages/ideal/</a></p>
</div>
<div id="ref-collado2016regionreport">
<p>20. Collado-Torres L, Jaffe AE, Leek JT. regionReport: Interactive reports for region-level and feature-level genomic analyses [version2; referees: 2 approved, 1 approved with reservations]. F1000Research. 2016 Jun;4:1–10. </p>
</div>
<div id="ref-robinson2010">
<p>21. Robinson MD, McCarthy DJ, Smyth GK. edgeR: a bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics (Oxford, England). 2010 Jan;26(1):139–40. </p>
</div>
<div id="ref-law2014voom">
<p>22. Law CW, Chen Y, Shi W, Smyth GK. Voom: Precision weights unlock linear model analysis tools for rna-seq read counts. Genome Biol. 2014;15(2):R29. </p>
</div>
<div id="ref-clusterProfiler">
<p>23. Yu G, Wang L-G, Han Y, He Q-Y. ClusterProfiler: An r package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012;16(5):284–7. </p>
</div>
<div id="ref-Soneson2015">
<p>24. Soneson C, Love MI, Robinson MD. Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences. [version 2; referees: 2 approved]. F1000Research. 2015;4(0):1521. </p>
</div>
<div id="ref-jaffe2018">
<p>25. Jaffe AE, Straub RE, Shin JH, Tao R, Gao Y, Collado-Torres L, et al. Developmental and genetic regulation of the human cortex transcriptome illuminate schizophrenia pathogenesis. Nature Neuroscience [Internet]. 2018 Aug [cited 2019 Apr 9];21(8):1117. Available from: <a href="https://www.nature.com/articles/s41593-018-0197-y">https://www.nature.com/articles/s41593-018-0197-y</a></p>
</div>
<div id="ref-epivizr">
<p>26. Bravo HC, Chelaru F, Smith L, Goldstein N, Kancherla J, Walter M, et al. Epivizr: R interface to epiviz web app. 2019. </p>
</div>
<div id="ref-wiggleplotr">
<p>27. Alasoo K. Wiggleplotr: Make read coverage plots from bigwig files. 2019. </p>
</div>
<div id="ref-jaffe2015">
<p>28. Jaffe AE, Shin J, Collado-Torres L, Leek JT, Tao R, Li C, et al. Developmental regulation of human cortex transcription and its clinical relevance at single base resolution. Nature Neuroscience. 2015 Jan;18(1):154–61. </p>
</div>
<div id="ref-bumphunter">
<p>29. Jaffe AE, Murakami P, Lee H, Leek JT, Fallin DM, Feinberg AP, et al. Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies. International journal of epidemiology. 2012;41(1):200–9. </p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
